diff --git a/lib/kolab_client_task.php b/lib/kolab_client_task.php
index 5320805..604a52c 100644
--- a/lib/kolab_client_task.php
+++ b/lib/kolab_client_task.php
@@ -859,6 +859,7 @@ class kolab_client_task
 
         case 'checkbox':
             $result['type'] = kolab_form::INPUT_CHECKBOX;
+            $result['checked'] = $data[$field['name']] == "TRUE";
             break;
 
         case 'password':
diff --git a/public_html/js/kolab_admin.js b/public_html/js/kolab_admin.js
index d7e4089..ab62c39 100644
--- a/public_html/js/kolab_admin.js
+++ b/public_html/js/kolab_admin.js
@@ -787,6 +787,11 @@ function kolab_admin()
       delete data.json[this.name + '-unit'];
     });
 
+    // checkbox inputs
+    $('input[type="checkbox"]', form).each(function() {
+      data.json[this.name] = this.checked?'TRUE':'FALSE';
+    });
+    
     return data;
   };
 
