diff --git a/lib/Kolab/FreeBusy/DirectoryLDAP.php b/lib/Kolab/FreeBusy/DirectoryLDAP.php
index f462f79..145cda9 100644
--- a/lib/Kolab/FreeBusy/DirectoryLDAP.php
+++ b/lib/Kolab/FreeBusy/DirectoryLDAP.php
@@ -99,9 +99,17 @@ class DirectoryLDAP extends Directory
                $result = array('s' => $user);
 
                if ($this->ready) {
+                        $base_dn = $config['base_dn'];
+ 
+                        if ($base_dn = '%dc') {
+                            list($u, $d) = explode('@', $user);
+
+                            $base_dn = 'dc='.strtr($d, array('.' => ',dc='));
+                        }
+
                        // search with configured filter
                        $filter = preg_replace('/%s/i', $user, $this->config['filter']);
-                       $ldapresult = $this->ldap->search($this->config['base_dn'], $filter, 'sub');
+                       $ldapresult = $this->ldap->search($base_dn, $filter, 'sub');
 
                        // got a valid result
                        if ($ldapresult && $ldapresult->count()) {

