diff --git a/lib/Auth.php b/lib/Auth.php
index f253185..cf79182 100644
--- a/lib/Auth.php
+++ b/lib/Auth.php
@@ -219,6 +219,11 @@ class Auth {
         return $this->auth_instance()->domainadmin_get_quota($domainadmin, $excludeuser);
     }
 
+    public function domainadmin_get_number_of_accounts($domainadmin)
+    {
+        return $this->auth_instance()->domainadmin_get_number_of_accounts($domainadmin);
+    }
+
     public function domain_edit($domain, $attributes, $typeid = null)
     {
         return $this->auth_instance()->domain_edit($domain, $attributes, $typeid);
diff --git a/lib/Auth/LDAP.php b/lib/Auth/LDAP.php
index db020bd..4b8c854 100644
--- a/lib/Auth/LDAP.php
+++ b/lib/Auth/LDAP.php
@@ -620,6 +620,28 @@ class LDAP extends Net_LDAP3 {
         return $quota;
     }
 
+    // get the number of accounts that this domainadmin manages.
+    public function domainadmin_get_number_of_accounts($domainadmin)
+    {
+        $numberOfAccounts = 0;
+        $domains = $this->domainadmin_get_domains($domainadmin);
+        foreach ($domains as $associateddomain) {
+            $domain = substr($associateddomain, strlen("associateddomain="));
+            $domain = substr($domain, 0, strpos($domain, ','));
+            // get all users that are part of this domain
+            $users_result = $this->_search(
+                               $this->_standard_root_dn($domain),
+                               "objectclass=kolabinetorgperson",
+                               array($this->conf->get("quota_attribute")));
+            if ($users_result != null && count($users_result) > 0) {
+               $users = $users_result->entries(true);
+               $numberOfAccounts += count($users);
+            }
+        }
+
+        return $numberOfAccounts;
+    }
+
     public function domain_edit($domain, $attributes, $typeid = null)
     {
         $domain = $this->domain_info($domain, array_keys($attributes));
--- a/lib/Auth/LDAP.php
+++ b/lib/Auth/LDAP.php
@@ -484,6 +484,20 @@
 
         $this->add_entry($dn, $attrs);
 
+        $dn = 'cn=enable-groupware-features,' . $inetdomainbasedn;
+        $attrs = array(
+            'objectclass' => array(
+                'top',
+                'ldapsubentry',
+                'nsroledefinition',
+                'nssimpleroledefinition',
+                'nsmanagedroledefinition',
+            ),
+            'cn' => 'enable-groupware-features'
+        );
+
+        $this->add_entry($dn, $attrs);
+
         $domain = $this->domain_info($domain, array_keys($attributes));
 
         if (empty($domain)) {
diff --git a/lib/api/kolab_api_service_domain.php b/lib/api/kolab_api_service_domain.php
index 4d5fcaa..92aaf4a 100644
--- a/lib/api/kolab_api_service_domain.php
+++ b/lib/api/kolab_api_service_domain.php
@@ -235,4 +235,60 @@ class kolab_api_service_domain extends kolab_api_service
 
         return false;
     }
+
+    /**
+     * get the default quota for a domain, specified in the TBits ISP LDAP attribute of the domain admin
+     *
+     * @param array $get   GET parameters
+     * @param array $post  POST parameters
+     *
+     * @return array|bool default quota, False on error
+     */
+    public function domain_defaultquota($getdata, $postdata)
+    {
+        Log::trace("domain.defaultquota(\$getdata = '" . var_export($getdata, TRUE) . "', \$postdata = '" . var_export($postdata, TRUE) . "')");
+
+        if (empty($getdata['id'])) {
+            Log::error("domain.defaultquota called without a Domain ID");
+            return false;
+        }
+
+        $auth   = Auth::get_instance();
+        $conf = Conf::get_instance();
+        $result = $auth->domain_info($getdata['id']);
+
+        // normalize result
+        $result = $this->parse_result_attributes('domain', $result);
+
+        if (empty($result['id'])) {
+            $result['id'] = $getdata['id'];
+        }
+
+        if ($result) {
+
+            // get domain admin for the current domain, and the default mailquota of the domain admin
+            unset($defaultdomainquota);
+            foreach ($result['domainadmin'] as $domainadmin) {
+                // ignore cn=Directory Manager
+                if ($domainadmin != $conf->get('bind_dn')) {
+                    $userinfo = array_shift(array_values($auth->user_info($domainadmin, array($quota_attribute))));
+                    if (isset($userinfo['tbitskolabdefaultquota'])) {
+                        // what about multiple domain admins of one domain?
+                        if (isset($defaultdomainquota)) {
+                              throw new Exception('error: only one domainadmin can have a default mailquota specified');
+                        }
+                        $defaultdomainquota = $userinfo['tbitskolabdefaultquota'];
+                    }
+                }
+            }
+
+            if (isset($defaultdomainquota)) {
+                $result['defaultquota'] = $defaultdomainquota;
+            }
+
+            return $result;
+        }
+        
+        return false;
+    }
 }
diff --git a/lib/api/kolab_api_service_form_value.php b/lib/api/kolab_api_service_form_value.php
index 82bde53..4d2a8cc 100644
--- a/lib/api/kolab_api_service_form_value.php
+++ b/lib/api/kolab_api_service_form_value.php
@@ -1219,6 +1219,61 @@ class kolab_api_service_form_value extends kolab_api_service
         return 'OK';
     }

+    private function validate_nsroledn($value, $postdata = array(), $validation_type = null)
+    {
+        foreach ($value as $element) {
+            if (strpos($element, "cn=enable-groupware-features") !== false) {
+                $conf = Conf::get_instance();
+                $auth = Auth::get_instance();
+
+                // get domain admin for the current domain, and the value of the flag tbitskolaballowgroupware
+                $domaindetails = array_shift(array_values($auth->domain_info($_SESSION['user']->get_domain(), array('domainadmin'))));
+                $domainadmins = $domaindetails['domainadmin'];
+                unset($tbitskolaballowgroupware);
+                foreach ($domaindetails['domainadmin'] as $domainadmin) {
+                    // ignore cn=Directory Manager
+                    if ($domainadmin != $conf->get('bind_dn')) {
+                        $userinfo = array_shift(array_values($auth->user_info($domainadmin, array("tbitskolaballowgroupware"))));
+                        if (isset($userinfo["tbitskolaballowgroupware"])) {
+                            // what about multiple domain admins of one domain?
+                            if (isset($tbitskolaballowgroupware) && $tbitskolaballowgroupware != $userinfo["tbitskolaballowgroupware"]) {
+                                  throw new Exception('error: domainadmins have contradicting settings for tbitskolaballowgroupware');
+                            }
+                            $tbitskolaballowgroupware = $userinfo["tbitskolaballowgroupware"];
+                        }
+                    }
+                }
+
+                if (isset($tbitskolaballowgroupware) && $tbitskolaballowgroupware == "FALSE") {
+                    throw new Exception('error: the domain admin does not have permissions to enable groupware features!');
+                }
+            }
+        }
+    }
+ 
+    private function validate_quota($value, $postdata = array(), $validation_type = null)
+    {
+        // convert MB/GB into KB
+        if (preg_match('/^([0-9]+)\s*(KB|MB|GB)$/i', $value, $m)) {
+            switch (strtoupper($m[2])) {
+            case 'KB': $value = $m[1]; break;
+            case 'MB': $value = $m[1] * 1024; break;
+            case 'GB': $value = $m[1] * 1024 * 1024; break;
+            }
+        }
+        return (string) intval($value);
+    }
+
+    private function validate_tbitskolaboverallquota($value, $postdata = array(), $validation_type = null)
+    {
+        return $this->validate_quota($value, $postdata, $validation_type);
+    }
+
+    private function validate_tbitskolabdefaultquota($value, $postdata = array(), $validation_type = null)
+    {
+        return $this->validate_quota($value, $postdata, $validation_type);
+    }
+ 
     private function validate_mailquota($value, $postdata = array(), $validation_type = null)
     {
         // convert MB/GB into KB
diff --git a/lib/api/kolab_api_service_user.php b/lib/api/kolab_api_service_user.php
index 7b6bd0e..593351f 100644
--- a/lib/api/kolab_api_service_user.php
+++ b/lib/api/kolab_api_service_user.php
@@ -70,6 +70,46 @@ class kolab_api_service_user extends kolab_api_service
     }
 
     /**
+     * check if the domain admin is allowed to add another account.
+     * using tbitsKolabMaxAccounts from LDAP
+     *
+     * @throws an exception if maximum number of accounts has been reached
+     */
+    private function validate_user_add()
+    {
+        $auth = Auth::get_instance();
+        $conf = Conf::get_instance();
+
+        // get domain admin for the current domain, and the mailquota of the domain admin
+        $domaindetails = array_shift(array_values($auth->domain_info($_SESSION['user']->get_domain(), array('domainadmin'))));
+        $domainadmins = $domaindetails['domainadmin'];
+        unset($domainadminmaxaccounts);
+        foreach ($domaindetails['domainadmin'] as $domainadmin) {
+            // ignore cn=Directory Manager
+            if ($domainadmin != $conf->get('bind_dn')) {
+                $userinfo = array_shift(array_values($auth->user_info($domainadmin, array($quota_attribute))));
+                if (isset($userinfo['tbitskolabmaxaccounts'])) {
+                    // what about multiple domain admins of one domain?
+                    if (isset($domainadminmaxaccounts)) {
+                          throw new Exception('error: only one domainadmin can have a maximum account number specified');
+                    }
+                    $domainadminmaxaccounts = $userinfo['tbitskolabmaxaccounts'];
+                    $maindomainadmin = $domainadmin;
+                }
+            }
+        }
+
+        if (isset($domainadminmaxaccounts)) {
+            $numberOfAccounts = $auth->domainadmin_get_number_of_accounts($maindomainadmin);
+            if ($numberOfAccounts >= $domainadminmaxaccounts) {
+                  throw new Exception('error: Cannot create another account.<br/>'.
+                      'maximum accounts booked: '.$domainadminmaxaccounts.'<br/>'.
+                      'Please order more accounts!');
+            }
+        }
+    }
+
+    /**
      * Create user.
      *
      * @param array $get   GET parameters
@@ -81,6 +121,9 @@ class kolab_api_service_user extends kolab_api_service
     {
         //console("user_add()", $postdata);
 
+        // check if the domainadmin is allowed to add more accounts
+        $this->validate_user_add();
+
         $user_attributes = $this->parse_input_attributes('user', $postdata);
 
         //console("user_add()", $user_attributes);
diff --git a/lib/client/kolab_client_task_user.php b/lib/client/kolab_client_task_user.php
index 40c2fcc..0100a83 100644
--- a/lib/client/kolab_client_task_user.php
+++ b/lib/client/kolab_client_task_user.php
@@ -88,6 +88,7 @@ class kolab_client_task_user extends kolab_client_task
             'contact_info'  => 'user.contact_info',
             'system'        => 'user.system',
             'config'        => 'user.config',
+            'domainadmin'   => 'user.domainadmin',
             'asterisk'      => 'user.asterisk',
             'other'         => 'user.other',
         );
@@ -143,6 +144,10 @@ class kolab_client_task_user extends kolab_client_task
             'kolabhomeserver'           => 'config',
             'mailhost'                  => 'config',
             'mailquota'                 => 'config',
+            'tbitskolabmaxaccounts'     => 'domainadmin',
+            'tbitskolaballowgroupware'  => 'domainadmin',
+            'tbitskolaboverallquota'    => 'domainadmin',
+            'tbitskolabdefaultquota'    => 'domainadmin',
             'cyrususerquota'            => 'config',
             'kolabfreebusyfuture'       => 'config',
             'kolabinvitationpolicy'     => 'config',
diff --git a/lib/kolab_api_service.php b/lib/kolab_api_service.php
index 8a64f3c..b4d4252 100644
--- a/lib/kolab_api_service.php
+++ b/lib/kolab_api_service.php
@@ -123,6 +123,10 @@ abstract class kolab_api_service
             return key($object_types);
         }
 
+        # sort object types by the key: Kolab user should win over Domain Admin, because it comes first
+        # there are only additional fields, so a kolab user would be displayed as admin, when sorting by type name
+        ksort($object_types);
+
         $object_class = array_map('strtolower', $object_class);
         $object_keys  = array_keys($attributes);
         $keys_count   = count($object_keys);
diff --git a/lib/kolab_client_task.php b/lib/kolab_client_task.php
index 8ba911b..cded524 100644
--- a/lib/kolab_client_task.php
+++ b/lib/kolab_client_task.php
@@ -645,6 +645,24 @@ class kolab_client_task
     }
 
     /**
+     * Returns the id of the type that is identified by the key
+     *
+     * @param string $types array of types for the current object class
+     * @param string $key identifies the type that we want the id for
+     *
+     * @return string the id for the type that we are looking for
+     */
+    protected function get_object_type_id_by_key($types, $key)
+    {
+       foreach ($types as $id => $data) {
+           if ($data['key'] == $key) {
+              return $id;
+           }
+       }
+       return -1;
+    }
+
+    /**
      * Returns user name.
      *
      * @param string $dn User DN attribute value
@@ -984,6 +1002,27 @@ class kolab_client_task
         if (!empty($data['type_id'])) {
             $type = $data['type_id'];
         }
+        else if ($name == "user") {
+            // set the default user type
+            $defaultUserType = $this->get_object_type_id_by_key($types, "kolab");
+            $conf = Conf::get_instance();
+            if ($conf->get('domainadmins_management_domain') == $_SESSION['user']['domain']) {
+                // in the management domain, default to DomainAdmin user
+                $defaultUserType = $this->get_object_type_id_by_key($types, "domainadmin");
+            } else {
+                // get the default mailquota of the domain admin
+                $result = $this->api_get('domain.defaultquota', array('id' => $_SESSION['user']['domain']));
+                $domaininfo = $result->get();
+
+                $defaultdomainquota = $domaininfo['defaultquota'];
+                if (isset($defaultdomainquota)) {
+                    // set the default mail quota
+                    $data['mailquota'] = $defaultdomainquota;
+                }
+            }
+
+            $data['type_id'] = $type = $defaultUserType;
+        }
         else {
             $data['type_id'] = $type = key($types);
         }
diff --git a/lib/locale/de_DE.php b/lib/locale/de_DE.php
index a728827..48e5d6a 100644
--- a/lib/locale/de_DE.php
+++ b/lib/locale/de_DE.php
@@ -239,6 +239,11 @@ $LANG['user.street'] = 'Straße';
 $LANG['user.system'] = 'System';
 $LANG['domain.admins'] = 'Domain Administratoren';
 $LANG['domain.domainadmin'] = 'Administratoren für diese Domain';
+$LANG['user.domainadmin'] = 'Domain Administrator';
+$LANG['user.tbitskolaballowgroupware'] = 'Erlaube Groupware Funktionen für Benutzer';
+$LANG['user.tbitskolabmaxaccounts'] = 'Maximale Anzahl von Benutzerkonten';
+$LANG['user.tbitskolaboverallquota'] = 'Gesamtquota verfügbar';
+$LANG['user.tbitskolabdefaultquota'] = 'Voreinstellung Quota für Benutzerkonten';
 $LANG['user.telephonenumber'] = 'Telefonnummer';
 $LANG['user.title'] = 'Jobbezeichnung';
 $LANG['user.type_id'] = 'Kontotyp';
diff --git a/lib/locale/en_US.php b/lib/locale/en_US.php
index abe885c..48b237e 100644
--- a/lib/locale/en_US.php
+++ b/lib/locale/en_US.php
@@ -309,6 +309,11 @@ $LANG['user.street'] = 'Street';
 $LANG['user.system'] = 'System';
 $LANG['domain.admins'] = 'Domain Administrators';
 $LANG['domain.domainadmin'] = 'Administrators for this domain';
+$LANG['user.domainadmin'] = 'Domain Administrator';
+$LANG['user.tbitskolaballowgroupware'] = 'Allow Groupware features for accounts';
+$LANG['user.tbitskolabmaxaccounts'] = 'Maximum number of accounts';
+$LANG['user.tbitskolaboverallquota'] = 'Overall Quota assigned';
+$LANG['user.tbitskolabdefaultquota'] = 'Default Quota for user accounts';
 $LANG['user.telephonenumber'] = 'Phone Number';
 $LANG['user.title'] = 'Job Title';
 $LANG['user.type_id'] = 'Account type';
