diff --git a/lib/Auth/LDAP.php b/lib/Auth/LDAP.php
index 4872521..5e11a35 100644
--- a/lib/Auth/LDAP.php
+++ b/lib/Auth/LDAP.php
@@ -339,6 +339,12 @@ class LDAP extends Net_LDAP3 {
             $service_bind_dn = $this->conf->get('ldap', 'bind_dn');
         }

+        // check if kolabInetOrgPerson is available
+        $self_attr = array('carLicense','description','displayName','facsimileTelephoneNumber','homePhone','homePostalAddress','initials','jpegPhoto','l','labeledURI','mobile','o','pager','photo','postOfficeBox','postalAddress','postalCode','preferredDeliveryMethod','preferredLanguage','registeredAddress','roomNumber','secretary','seeAlso','st','street','telephoneNumber','telexNumber','title','userCertificate','userPassword','userSMIMECertificate','x500UniqueIdentifier');
+        if(in_array('kolabInetOrgPerson', $this->classes_allowed())) {
+            $self_attr = array_merge($self_attr, array('kolabDelegate','kolabInvitationPolicy','kolabAllowSMTPSender'));
+        }
+
         $dn = $inetdomainbasedn;
         $attrs = array(
                 // @TODO: Probably just use ldap_explode_dn()
@@ -349,7 +355,7 @@ class LDAP extends Net_LDAP3 {
                     ),
                 'aci' => array(
                         // Self-modification
-                        "(targetattr=\"carLicense || description || displayName || facsimileTelephoneNumber || homePhone || homePostalAddress || initials || jpegPhoto || labeledURI || mobile || pager || photo || postOfficeBox || postalAddress || postalCode || preferredDeliveryMethod || preferredLanguage || registeredAddress || roomNumber || secretary || seeAlso || st || street || telephoneNumber || telexNumber || title || userCertificate || userPassword || userSMIMECertificate || x500UniqueIdentifier\")(version 3.0; acl \"Enable self write for common attributes\"; allow (write) userdn=\"ldap:///self\";)",
+                        "(targetattr=\"".implode(" || ",$self_attr)."\")(version 3.0; acl \"Enable self write for common attributes\"; allow (write) userdn=\"ldap:///self\";)",

                         // Directory Administrators
                         "(targetattr =\"*\")(version 3.0;acl \"Directory Administrators Group\";allow (all) (groupdn=\"ldap:///cn=Directory Administrators," . $inetdomainbasedn . "\" or roledn=\"ldap:///cn=kolab-admin," . $inetdomainbasedn . "\");)",