From 8474639ef74f9fdb00759208b27d970429338a8a Mon Sep 17 00:00:00 2001
From: Daniel Hoffend <dh@dotlan.net>
Date: Fri, 01 Nov 2013 22:57:28 +0000
Subject: Fix #2466 - List removes last input field when other similar input fields exists in the form.

---
diff --git a/public_html/js/kolab_admin.js b/public_html/js/kolab_admin.js
index e35e43b..ab8dec8 100644
--- a/public_html/js/kolab_admin.js
+++ b/public_html/js/kolab_admin.js
@@ -935,7 +935,7 @@ function kolab_admin()
       $('span[class="reset"]', elem).click(function() {
         var span = $(this.parentNode.parentNode),
           name = data.name.replace(/\[[0-9]+\]$/, ''),
-          l = $('input[name^="' + name + '"]', form),
+          l = $('input[name^="' + name + '["]', form),
           key = $(this).data('key');
 
         if (l.length > 1 || $('input[name="' + name + '"]', form).attr('data-autocomplete'))
--
cgit v0.8.2.1
