diff --git a/pykolab/imap/__init__.py b/pykolab/imap/__init__.py
index ac359a0..23bf4a9 100644
--- a/pykolab/imap/__init__.py
+++ b/pykolab/imap/__init__.py
@@ -588,6 +588,7 @@ class IMAP(object):
                 folder_name = "%s%s" % (personal, folder_name)
 
             try:
+                self.create_folder(folder_name)
                 created = False
                 last_log = time.time()
                 while not created:

