diff --git a/auth/ldap/__init__.py b/auth/ldap/__init__.py
index 535d360..1b6a5a8 100644
--- a/auth/ldap/__init__.py
+++ b/auth/ldap/__init__.py
@@ -727,7 +727,10 @@ class LDAP(pykolab.base.Base):
             entry['preferredlanguage'] = conf.get('kolab', 'default_locale')
 
         # Primary mail address
-        if not primary_mail == None:
+        # Patch by tbits: only apply primary mail address policy if the mail address does not end with the proper domain name
+        if not primary_mail == None and (entry.has_key(primary_mail_attribute) and entry[primary_mail_attribute].endswith(self.domain)):
+            primary_mail_address = entry[primary_mail_attribute]
+        elif not primary_mail == None:
             primary_mail_address = conf.plugins.exec_hook(
                     "set_primary_mail",
                     kw={
