--- a/pykolab/setup/setup_mta.py	2016-12-14 10:09:06.163000000 +0000
+++ b/pykolab/setup/setup_mta.py	2016-12-14 10:11:41.832000000 +0000
@@ -267,7 +267,6 @@
             "submission_recipient_restrictions": "check_policy_service unix:private/submission_policy, permit_sasl_authenticated, reject",
             "submission_sender_restrictions": "reject_non_fqdn_sender, check_policy_service unix:private/submission_policy, permit_sasl_authenticated, reject",
             "submission_data_restrictions": "check_policy_service unix:private/submission_policy",
-            "content_filter": "smtp-amavis:[127.0.0.1]:10024"
 
         }
 
@@ -498,36 +497,24 @@
 
     if os.path.isfile('/bin/systemctl'):
         subprocess.call(['systemctl', 'restart', 'postfix.service'])
-        subprocess.call(['systemctl', 'restart', amavisservice])
-        subprocess.call(['systemctl', 'restart', clamavservice])
         subprocess.call(['systemctl', 'restart', 'wallace.service'])
     elif os.path.isfile('/sbin/service'):
         subprocess.call(['service', 'postfix', 'restart'])
-        subprocess.call(['service', 'amavisd', 'restart'])
-        subprocess.call(['service', 'clamd.amavisd', 'restart'])
         subprocess.call(['service', 'wallace', 'restart'])
     elif os.path.isfile('/usr/sbin/service'):
         subprocess.call(['/usr/sbin/service','postfix','restart'])
-        subprocess.call(['/usr/sbin/service','amavis','restart'])
-        subprocess.call(['/usr/sbin/service','clamav-daemon','restart'])
         subprocess.call(['/usr/sbin/service','wallace','restart'])
     else:
         log.error(_("Could not start the postfix, clamav and amavisd services services."))
 
     if os.path.isfile('/bin/systemctl'):
         subprocess.call(['systemctl', 'enable', 'postfix.service'])
-        subprocess.call(['systemctl', 'enable', amavisservice])
-        subprocess.call(['systemctl', 'enable', clamavservice])
         subprocess.call(['systemctl', 'enable', 'wallace.service'])
     elif os.path.isfile('/sbin/chkconfig'):
         subprocess.call(['chkconfig', 'postfix', 'on'])
-        subprocess.call(['chkconfig', 'amavisd', 'on'])
-        subprocess.call(['chkconfig', 'clamd.amavisd', 'on'])
         subprocess.call(['chkconfig', 'wallace', 'on'])
     elif os.path.isfile('/usr/sbin/update-rc.d'):
         subprocess.call(['/usr/sbin/update-rc.d', 'postfix', 'defaults'])
-        subprocess.call(['/usr/sbin/update-rc.d', 'amavis', 'defaults'])
-        subprocess.call(['/usr/sbin/update-rc.d', 'clamav-daemon', 'defaults'])
         subprocess.call(['/usr/sbin/update-rc.d', 'wallace', 'defaults'])
     else:
         log.error(_("Could not configure to start on boot, the " + \

