--- a/lib/kolab_client_task.php.orig	2014-12-01 19:00:36.913301999 +0100
+++ b/lib/kolab_client_task.php	2014-12-01 19:09:22.249301999 +0100
@@ -1042,6 +1046,18 @@
             reset($types);
 
             $data['type_id'] = $type = ($default !== null ? $default : key($types));
+
+            if ($name == "user") {
+                // get the default mailquota of the domain admin
+                $result = $this->api_get('domain.domainadmin_info', array('variablename' => 'tbitskolabdefaultquota'));
+                $domaininfo = $result->get();
+
+                $defaultdomainquota = $domaininfo['tbitskolabdefaultquota'];
+                if (isset($defaultdomainquota)) {
+                    // set the default mail quota
+                    $data['mailquota'] = $defaultdomainquota;
+                }
+            }
         }
 
         if ($type) {
