diff --git a/lib/kolab_client_task.php b/lib/kolab_client_task.php
index 61023df..5206a50 100644
--- a/lib/kolab_client_task.php
+++ b/lib/kolab_client_task.php
@@ -1050,6 +1050,18 @@ class kolab_client_task
             reset($types);
 
             $data['type_id'] = $type = ($default !== null ? $default : key($types));
+
+            if ($name == "user") {
+                // get the default mailquota of the domain admin
+                $result = $this->api_get('domain.domainadmin_info', array('variablename' => 'tbitskolabdefaultquota'));
+                $domaininfo = $result->get();
+
+                $defaultdomainquota = $domaininfo['tbitskolabdefaultquota'];
+                if (isset($defaultdomainquota)) {
+                    // set the default mail quota
+                    $data['mailquota'] = $defaultdomainquota;
+                }
+            }
         }
 
         if ($type) {
