--- a/lib/kolab_client_task.php	2014-12-01 11:19:31.981301999 +0100
+++ b/lib/kolab_client_task.php	2014-12-01 11:16:16.993301999 +0100
@@ -1046,9 +1046,19 @@
             // set the default user type
             $defaultUserType = $this->get_object_type_id_by_key($types, "kolab");
             $data['type_id'] = $type = $defaultUserType;
+
+            // get the default mailquota of the domain admin
+            $result = $this->api_get('domain.domainadmin_info', array('variablename' => 'tbitskolabdefaultquota'));
+            $domaininfo = $result->get();
+
+            $defaultdomainquota = $domaininfo['tbitskolabdefaultquota'];
+            if (isset($defaultdomainquota)) {
+                // set the default mail quota
+                $data['mailquota'] = $defaultdomainquota;
+            }
         }
         else {
             $data['type_id'] = $type = key($types);
         }
 
         if ($type) {
