--- a/share/templates/amavisd.conf.tpl.orig
+++ b/share/kolab/templates/amavisd.conf.tpl
@@ -373,7 +373,7 @@ use strict;
 
 # \#\## http://www.clamav.net/
 ['ClamAV-clamd',
-  \&ask_daemon, ["CONTSCAN {}\n", "/var/spool/amavisd/clamd.sock"],
+  \&ask_daemon, ["CONTSCAN {}\n", "$clamdsock"],
   qr/\bOK\$/m, qr/\bFOUND\$/m,
   qr/^.*?: (?!Infected Archive)(.*) FOUND\$/m ],
 # # NOTE: run clamd under the same user as amavisd, or run it under its own
