--- a/Net/LDAP3.php	2015-06-24 10:34:41.913985719 +0200
+++ b/Net/LDAP3.php	2015-06-24 10:35:08.197984491 +0200
@@ -665,7 +665,7 @@
     public function delete_entry_recursive($entry_dn)
     {
         // searching for sub entries, but not scope sub, just one level
-        $result = $this->search($entry_dn, '(objectclass=*)', 'one');
+        $result = $this->search($entry_dn, '(|(objectclass=*)(objectclass=ldapsubentry))', 'one');
 
         if ($result) {
             $entries = $result->entries(true);
