--- a/public_html/js/kolab_admin.js	2014-02-11 11:37:44.000000000 +0100
+++ b/public_html/js/kolab_admin.js	2014-07-11 14:44:12.623960515 +0200
@@ -808,6 +808,8 @@
       var unit = $('select[name="' + this.name + '-unit"]').val();
       if (unit && this.value)
         data.json[this.name] = this.value + unit;
+      if (!this.value)
+        data.json[this.name] = "0kb";
       delete data.json[this.name + '-unit'];
     });
 
--- a/lib/kolab_html.php	2014-02-11 11:37:44.000000000 +0100
+++ b/lib/kolab_html.php	2014-07-11 14:36:42.059959931 +0200
@@ -182,6 +182,7 @@
             if ($unit) {
                 $attribs['value'] .= ' ' . strtoupper($unit);
             }
+            $attribs['data-type'] = 'quota';
 
             return self::input($attribs);
         }
