--- a/pykolab/auth/ldap/__init__.py	2014-07-14 13:51:25.819960000 +0200
+++ b/pykolab/auth/ldap/__init__.py	2014-07-14 13:51:05.691960012 +0200
@@ -2194,7 +2194,7 @@
            The exact form and shape of the feedback very much depends on the
            supportedControl used to even get the data.
         """
-
+        entry = None
         log.debug(
                 "auth.ldap.LDAP._synchronize_callback(args %r, kw %r)" % (
                         args,
@@ -2296,6 +2296,8 @@
 #                        )
 #
 #                    server = self.imap.user_mailbox_server(folder)
+        if entry == None:
+          return
         log.debug(
                 _("Done with _synchronize_callback() for entry %r") % (
                         entry['id']
