diff --git a/lib/Kolab/FreeBusy/DirectoryLDAP.php b/lib/Kolab/FreeBusy/DirectoryLDAP.php
--- a/lib/Kolab/FreeBusy/DirectoryLDAP.php
+++ b/lib/Kolab/FreeBusy/DirectoryLDAP.php
@@ -77,10 +77,18 @@
 	{
 		$result = array('s' => $user);
 
+                $base_dn = $config['base_dn'];
+ 
+                if ($base_dn = '%dc') {
+                    list($u, $d) = explode('@', $user);
+
+                    $base_dn = 'dc='.strtr($d, array('.' => ',dc='));
+                }
+
 		if ($this->ready) {
 			// search with configured filter
 			$filter = preg_replace('/%s/i', $user, $this->config['filter']);
-			$ldapresult = $this->ldap->search($this->config['base_dn'], $filter, 'sub');
+			$ldapresult = $this->ldap->search($base_dn, $filter, 'sub');
 
 			// got a valid result
 			if ($ldapresult && $ldapresult->count()) {
