--- LDAP.orig	2017-06-07 14:49:10.525162243 +0200
+++ /usr/share/kolab-webadmin/lib/Auth/LDAP.php	2017-06-07 14:49:44.736163982 +0200
@@ -271,7 +271,7 @@
         $admin_readonly_attrs = array("tbitsKolabMaxAccounts", "tbitsKolabOverallQuota");
 
         if (in_array('tbitsKolabUser', $this->classes_allowed())) {
-            $self_attrs = array_merge($self_attrs, array('tbitsKolabLastLogin', 'tbitsKolabQuotaUsed'));
+            $self_attrs = array_merge($self_attrs, array('tbitsKolabLastLogin', 'tbitsKolabQuotaUsed', 'tbitsKolabIntranetToken'));
         }
 
         $_domain = str_replace('.', '_', $domain);
--- en_US.php	2017-06-07 15:00:24.488196507 +0200
+++ //usr/share/kolab-webadmin/lib/locale/en_US.php	2017-06-07 15:01:16.455199149 +0200
@@ -451,6 +451,7 @@
 $LANG['user.tbitskolaboverallquota'] = 'Overall Quota assigned';
 $LANG['user.tbitskolabdefaultquota'] = 'Default Quota for user accounts';
 $LANG['user.tbitskolablastlogin'] = 'Latest successful login';
+$LANG['user.tbitskolabintranettoken'] = 'Intranet token';
 $LANG['user.tbitskolabquotaused'] = 'Current quota usage';
 $LANG['quota.unlimited'] = "Unlimited";
 $LANG['user.title'] = 'Job Title';
--- de_DE.php	2017-06-07 15:01:22.184199440 +0200
+++ /usr/share/kolab-webadmin/lib/locale/de_DE.php	2017-06-07 15:01:49.481200828 +0200
@@ -464,6 +464,7 @@
 $LANG['user.tbitskolabdefaultquota'] = 'Voreinstellung Quota für Benutzerkonten';
 $LANG['user.statistics'] = 'Info';
 $LANG['user.tbitskolablastlogin'] = 'Letzte erfolgreiche Anmeldung';
+$LANG['user.tbitskolabintranettoken'] = 'Intranet Anmeldung';
 $LANG['user.tbitskolabquotaused'] = 'Aktueller Speicherplatzverbrauch';
 $LANG['quota.unlimited'] = "Unbegrenzt";
 
--- kolab_client_task_user.php	2017-06-07 15:03:00.731204450 +0200
+++ /usr/share/kolab-webadmin/lib/client/kolab_client_task_user.php	2017-06-07 15:21:12.860259972 +0200
@@ -171,6 +171,7 @@
             'uid'                       => 'system',
             'userpassword'              => 'system',
             'userpassword2'             => 'system',
+            'tbitskolabintranettoken'   => 'system',
             'uidnumber'                 => 'system',
             'gidnumber'                 => 'system',
             'homedirectory'             => 'system',
@@ -220,6 +221,14 @@
         $form = $this->form_prepare('user', $data, array('userpassword2'), null, $fields_map['type_id']);
         list($fields, $types, $type, $add_mode) = $form;
 
+        // check if tbitsKolabIntranetToken should be active for the current domain
+        // see kolab.conf, [kolab] enable_intranet_token, which is a comma separated list of domains
+        $conf = Conf::get_instance();
+        $domainsWithIntranetToken = explode(',', $conf->get('enable_intranet_token'));
+        if (!in_array($_SESSION['user']['domain'], $domainsWithIntranetToken)) {
+            unset($fields['tbitskolabintranettoken']);
+        }
+
         // Add password confirmation
         if (isset($fields['userpassword'])) {
             $fields['userpassword2'] = $fields['userpassword'];
