--- a/saslauthd/__init__.py
+++ b/saslauthd/__init__.py
@@ -170,13 +170,13 @@ class SASLAuthDaemon(object):
 
         # TODO: The saslauthd socket path could be a setting.
         try:
-            os.remove(socketfile)
+            os.remove(conf.socketfile)
         except:
             # TODO: Do the "could not remove, could not start" dance
             pass
 
-        s.bind(socketfile)
-        os.chmod(socketfile, 0777)
+        s.bind(conf.socketfile)
+        os.chmod(conf.socketfile, 0777)
 
         s.listen(5)
 
@@ -271,7 +271,7 @@ class SASLAuthDaemon(object):
 
     def _ensure_socket_dir(self):
         utils.ensure_directory(
-                os.path.dirname(socketfile),
+                os.path.dirname(conf.socketfile),
                 conf.process_username,
                 conf.process_groupname
             )

