diff --git a/lib/Auth/LDAP.php b/lib/Auth/LDAP.php
index aafe618..abd33eb 100644
--- a/lib/Auth/LDAP.php
+++ b/lib/Auth/LDAP.php
@@ -251,6 +251,10 @@ class LDAP extends Net_LDAP3 {
              $self_attrs = array_merge($self_attrs, array('kolabDelegate', 'kolabInvitationPolicy', 'kolabAllowSMTPSender'));
         }
 
+        if (in_array('tbitsKolabUser', $this->classes_allowed())) {
+            $self_attrs = array_merge($self_attrs, array('tbitsKolabLastLogin'));
+        }
+
         $_domain = str_replace('.', '_', $domain);
         $dn      = $inetdomainbasedn;
         $cn      = str_replace(array(',', '='), array('\2C', '\3D'), $dn);
diff --git a/lib/client/kolab_client_task_settings.php b/lib/client/kolab_client_task_settings.php
index 9a23161..1f16029 100644
--- a/lib/client/kolab_client_task_settings.php
+++ b/lib/client/kolab_client_task_settings.php
@@ -34,6 +34,7 @@ class kolab_client_task_settings extends kolab_client_task
     protected $form_element_types = array(
         'text', 'text-separated', 'text-quota', 'text-autocomplete',
+        'text-unixtimestamp',
         'select', 'multiselect',
         'list', 'list-autocomplete', 'checkbox', 'password', 'ldap_url',
         'aci', 'imap_acl',
     );
diff --git a/lib/client/kolab_client_task_user.php b/lib/client/kolab_client_task_user.php
index c5d71d9..4739dcd 100644
--- a/lib/client/kolab_client_task_user.php
+++ b/lib/client/kolab_client_task_user.php
@@ -89,6 +89,7 @@ class kolab_client_task_user extends kolab_client_task
             'system'        => 'user.system',
             'config'        => 'user.config',
             'domainadmin'   => 'user.domainadmin',
+            'statistics'    => 'user.statistics',
             'asterisk'      => 'user.asterisk',
             'other'         => 'user.other',
         );
@@ -128,6 +129,8 @@ class kolab_client_task_user extends kolab_client_task
             'alias'                     => 'contact_info',
             'mailalternateaddress'      => 'contact_info',
 
+            'tbitskolablastlogin'       => 'statistics',
+
             /* POSIX Attributes first */
             'uid'                       => 'system',
             'userpassword'              => 'system',
diff --git a/lib/kolab_client_task.php b/lib/kolab_client_task.php
index 813fa02..e8f4d0f 100644
--- a/lib/kolab_client_task.php
+++ b/lib/kolab_client_task.php
@@ -892,6 +892,10 @@ class kolab_client_task
             }
             break;
 
+        case 'text-unixtimestamp':
+            $result['type']    = kolab_form::INPUT_TEXTUNIXTIMESTAMP;
+            break;
+
         case 'text-quota':
             $result['type']    = kolab_form::INPUT_TEXTQUOTA;
             $result['default'] = $field['default'];
diff --git a/lib/kolab_form.php b/lib/kolab_form.php
index 1c26be7..7fb2bf7 100644
--- a/lib/kolab_form.php
+++ b/lib/kolab_form.php
@@ -40,5 +40,6 @@ class kolab_form
     const INPUT_CONTENT  = 20;
     const INPUT_TEXTQUOTA = 30;
+    const INPUT_TEXTUNIXTIMESTAMP = 40;
 
     private $attribs  = array();
     private $elements = array();
@@ -282,6 +283,11 @@ class kolab_form
             $content = kolab_html::inputquota($attribs);
             break;
 
+        case self::INPUT_TEXTUNIXTIMESTAMP:
+            $attribs['type'] = 'text';
+            $content = kolab_html::inputunixtimestamp($attribs);
+            break;
+
         case self::INPUT_CHECKBOX:
             $attribs['type'] = 'checkbox';
             $content = kolab_html::input($attribs);
diff --git a/lib/kolab_html.php b/lib/kolab_html.php
index 8a9243f..1585111 100644
--- a/lib/kolab_html.php
+++ b/lib/kolab_html.php
@@ -206,6 +206,28 @@ class kolab_html
     }
 
     /**
+     * Readonly control that will display the time encoded as a unix timestamp. Used for displaying tbitsKolabLastLogin
+     *
+     * @param array  $attribs  Element attributes
+     *
+     * @return string HTML output of the timestamp
+     */
+    public static function inputunixtimestamp($attribs = array())
+    {
+        $attribs['type'] = 'hidden';
+        $hidden_input = self::input($attribs);
+        unset($attribs['type']);
+        $attribs['name'] .= "_display";
+        $attribs['readonly'] = true;
+        $attribs['disabled'] = true;
+	if (!empty($attribs['value'])) {
+            $attribs['value'] = date('d.m.Y H:i:s e', $attribs['value']);
+        }
+        $readonly_input = self::input($attribs);
+        return $hidden_input.$readonly_input;
+    }
+
+    /**
      * Textarea element.
      *
      * @param array $attribs  Element attributes
diff --git a/lib/locale/de_DE.php b/lib/locale/de_DE.php
index 1609cc3..e806293 100644
--- a/lib/locale/de_DE.php
+++ b/lib/locale/de_DE.php
@@ -448,5 +448,7 @@
 $LANG['domain.domainadmin'] = 'Administratoren für diese Domain';
 $LANG['user.tbitskolabmaxaccounts'] = 'Maximale Anzahl von Benutzerkonten';
 $LANG['user.tbitskolaboverallquota'] = 'Gesamtquota verfügbar';
 $LANG['user.tbitskolabdefaultquota'] = 'Voreinstellung Quota für Benutzerkonten';
+$LANG['user.statistics'] = 'Info';
+$LANG['user.tbitskolablastlogin'] = 'Letzte erfolgreiche Anmeldung';

diff --git a/lib/locale/en_US.php b/lib/locale/en_US.php
index fbb51c9..4bf06d3 100644
--- a/lib/locale/en_US.php
+++ b/lib/locale/en_US.php
@@ -424,12 +424,14 @@ $LANG['user.postcode'] = 'Postal code';
 $LANG['user.preferredlanguage'] = 'Native tongue';
 $LANG['user.room'] = 'Room number';
 $LANG['user.sn'] = 'Surname';
+$LANG['user.statistics'] = 'Info';
 $LANG['user.street'] = 'Street';
 $LANG['user.system'] = 'System';
 $LANG['user.telephonenumber'] = 'Phone Number';
 $LANG['user.tbitskolabmaxaccounts'] = 'Maximum number of accounts';
 $LANG['user.tbitskolaboverallquota'] = 'Overall Quota assigned';
 $LANG['user.tbitskolabdefaultquota'] = 'Default Quota for user accounts';
+$LANG['user.tbitskolablastlogin'] = 'Latest successful login';
 $LANG['user.title'] = 'Job Title';
 $LANG['user.type_id'] = 'Account type';
 $LANG['user.uid'] = 'Unique identity (UID)';
