diff --git a/lib/Auth/LDAP.php b/lib/Auth/LDAP.php
index bf02ba1..ccd0a87 100644
--- a/lib/Auth/LDAP.php
+++ b/lib/Auth/LDAP.php
@@ -373,6 +373,10 @@ class LDAP extends Net_LDAP3 {
              $self_attrs = array_merge($self_attrs, array('kolabDelegate', 'kolabInvitationPolicy', 'kolabAllowSMTPSender'));
         }
 
+        if (in_array('tbitsKolabUser', $this->classes_allowed())) {
+            $self_attrs = array_merge($self_attrs, array('tbitsKolabLastLogin'));
+        }
+
         $attrs = array(
             // @TODO: Probably just use ldap_explode_dn()
             'dc' => substr($dn, (strpos($dn, '=')+1), ((strpos($dn, ',')-strpos($dn, '='))-1)),

diff -uNr a/lib/client/kolab_client_task_settings.php b/lib/client/kolab_client_task_settings.php
--- a/lib/client/kolab_client_task_settings.php	2013-11-26 11:03:49.476153091 +0100
+++ b/lib/client/kolab_client_task_settings.php	2013-11-26 11:02:24.154153091 +0100
@@ -34,6 +34,7 @@
     protected $form_element_types = array(
         'text', 'select', 'multiselect', 'list', 'list-autocomplete', 'checkbox', 'password', 'ldap_url',
         'text-quota',
+        'text-unixtimestamp',
         'list-domainadmins',
     );
 
diff -uNr a/lib/client/kolab_client_task_user.php b/lib/client/kolab_client_task_user.php
--- a/lib/client/kolab_client_task_user.php	2013-11-26 11:03:49.566153091 +0100
+++ b/lib/client/kolab_client_task_user.php	2013-11-26 11:02:24.154153091 +0100
@@ -89,6 +89,7 @@
             'system'        => 'user.system',
             'config'        => 'user.config',
             'domainadmin'   => 'user.domainadmin',
+            'statistics'    => 'user.statistics',
             'asterisk'      => 'user.asterisk',
             'other'         => 'user.other',
         );
@@ -128,6 +129,8 @@
             'alias'                     => 'contact_info',
             'mailalternateaddress'      => 'contact_info',
 
+            'tbitskolablastlogin'       => 'statistics',
+
             /* POSIX Attributes first */
             'uid'                       => 'system',
             'userpassword'              => 'system',
diff -uNr a/lib/kolab_client_task.php b/lib/kolab_client_task.php
--- a/lib/kolab_client_task.php	2013-11-26 11:03:49.539153091 +0100
+++ b/lib/kolab_client_task.php	2013-11-26 11:02:24.153153091 +0100
@@ -892,6 +892,10 @@
             }
             break;
 
+        case 'text-unixtimestamp':
+            $result['type']    = kolab_form::INPUT_TEXTUNIXTIMESTAMP;
+            break;
+
         case 'text-quota':
             $result['type']    = kolab_form::INPUT_TEXTQUOTA;
             $result['default'] = $field['default'];
diff -uNr a/lib/kolab_form.php b/lib/kolab_form.php
--- a/lib/kolab_form.php	2013-11-26 11:03:49.478153091 +0100
+++ b/lib/kolab_form.php	2013-11-26 11:02:24.154153091 +0100
@@ -40,6 +40,7 @@
     const INPUT_CONTENT  = 20;
     const INPUT_TEXTQUOTA = 30;
     const INPUT_DOMAINADMIN = 40;
+    const INPUT_TEXTUNIXTIMESTAMP = 50;
 
     const TYPE_LIST = 1;
 
@@ -284,6 +285,11 @@
             $content = kolab_html::inputquota($attribs);
             break;
 
+        case self::INPUT_TEXTUNIXTIMESTAMP:
+            $attribs['type'] = 'text';
+            $content = kolab_html::inputunixtimestamp($attribs);
+            break;
+
         case self::INPUT_CHECKBOX:
             $attribs['type'] = 'checkbox';
             $content = kolab_html::input($attribs);
diff -uNr a/lib/kolab_html.php b/lib/kolab_html.php
--- a/lib/kolab_html.php	2013-11-24 14:44:37.000000000 +0100
+++ b/lib/kolab_html.php	2013-11-26 11:02:24.155153091 +0100
@@ -205,6 +205,28 @@
     }
 
     /**
+     * Readonly control that will display the time encoded as a unix timestamp. Used for displaying tbitsKolabLastLogin
+     *
+     * @param array  $attribs  Element attributes
+     *
+     * @return string HTML output of the timestamp
+     */
+    public static function inputunixtimestamp($attribs = array())
+    {
+        $attribs['type'] = 'hidden';
+        $hidden_input = self::input($attribs);
+        unset($attribs['type']);
+        $attribs['name'] .= "_display";
+        $attribs['readonly'] = true;
+        $attribs['disabled'] = true;
+	if (!empty($attribs['value'])) {
+            $attribs['value'] = date('d.m.Y H:i:s e', $attribs['value']);
+        }
+        $readonly_input = self::input($attribs);
+        return $hidden_input.$readonly_input;
+    }
+
+    /**
      * Textarea element.
      *
      * @param array $attribs  Element attributes
diff -uNr a/lib/locale/de_DE.php b/lib/locale/de_DE.php
--- a/lib/locale/de_DE.php	2013-11-26 11:03:49.479153091 +0100
+++ b/lib/locale/de_DE.php	2013-11-26 11:02:24.157153091 +0100
@@ -239,12 +239,14 @@
 $LANG['user.room'] = 'Raumnummer';
 $LANG['user.sn'] = 'Nachname';
 $LANG['user.street'] = 'Straße';
+$LANG['user.statistics'] = 'Info';
 $LANG['user.system'] = 'System';
 $LANG['user.telephonenumber'] = 'Telefonnummer';
 $LANG['user.tbitskolaballowgroupware'] = 'Erlaube Groupware Funktionen für Benutzer';
 $LANG['user.tbitskolabmaxaccounts'] = 'Maximale Anzahl von Benutzerkonten';
 $LANG['user.tbitskolaboverallquota'] = 'Gesamtquota verfügbar';
 $LANG['user.tbitskolabdefaultquota'] = 'Voreinstellung Quota für Benutzerkonten';
+$LANG['user.tbitskolablastlogin'] = 'Letzte erfolgreiche Anmeldung';
 $LANG['user.title'] = 'Jobbezeichnung';
 $LANG['user.type_id'] = 'Kontotyp';
 $LANG['user.uid'] = 'Eindeutige Identität (UID)';
diff -uNr a/lib/locale/en_US.php b/lib/locale/en_US.php
--- a/lib/locale/en_US.php	2013-11-26 11:03:49.479153091 +0100
+++ b/lib/locale/en_US.php	2013-11-26 11:02:24.157153091 +0100
@@ -315,6 +315,7 @@
 $LANG['user.preferredlanguage'] = 'Native tongue';
 $LANG['user.room'] = 'Room number';
 $LANG['user.sn'] = 'Surname';
+$LANG['user.statistics'] = 'Info';
 $LANG['user.street'] = 'Street';
 $LANG['user.system'] = 'System';
 $LANG['user.telephonenumber'] = 'Phone Number';
@@ -322,6 +323,7 @@
 $LANG['user.tbitskolabmaxaccounts'] = 'Maximum number of accounts';
 $LANG['user.tbitskolaboverallquota'] = 'Overall Quota assigned';
 $LANG['user.tbitskolabdefaultquota'] = 'Default Quota for user accounts';
+$LANG['user.tbitskolablastlogin'] = 'Latest successful login';
 $LANG['user.title'] = 'Job Title';
 $LANG['user.type_id'] = 'Account type';
 $LANG['user.uid'] = 'Unique identity (UID)';

