diff --git a/lib/Auth.php b/lib/Auth.php
index 591dd3a..725c2d4 100644
--- a/lib/Auth.php
+++ b/lib/Auth.php
@@ -309,6 +309,11 @@ class Auth {
         return $this->auth_instance($domain)->list_users($attributes, $search, $params);
     }
 
+    public function list_admins($domain = NULL, $attributes = array())
+    {
+        return $this->auth_instance($domain)->list_admins($attributes);
+    }
+
     public function list_groups($domain = NULL, $attributes = array(), $search = array(), $params = array())
     {
         return $this->auth_instance($domain)->list_groups($attributes, $search, $params);
diff --git a/lib/Auth/LDAP.php b/lib/Auth/LDAP.php
index e4a919e..28e07bf 100644
--- a/lib/Auth/LDAP.php
+++ b/lib/Auth/LDAP.php
@@ -24,6 +24,7 @@
 */
 
 require_once "Net/LDAP3.php";
+require_once "/usr/share/roundcubemail/plugins/managesieve/lib/Roundcube/rcube_sieve_script.php";
 
 /**
  * Kolab LDAP handling abstraction class.
@@ -651,6 +652,16 @@ class LDAP extends Net_LDAP3 {
             $result = $this->legacy_rights($subject);
         }
 
+        if ($_SESSION['user']->user_bind_dn == "cn=Directory Manager") {
+           $result['entryLevelRights'][] = "report";
+        } else {
+            $user = $this->user_info($_SESSION['user']->user_bind_dn, array('objectclass', 'nsrole'));
+            if ((!empty($user[$_SESSION['user']->user_bind_dn]['nsrole']) && strpos($user[$_SESSION['user']->user_bind_dn]['nsrole'], '=kolab-admin,dc=') > 0) ||
+                (in_array('tbitskolabdomainadmin', array_values($user[$_SESSION['user']->user_bind_dn]['objectclass'])))) {
+                $result['entryLevelRights'][] = "report";
+            }
+        }
+
         if (!$this->set_cache_data($ckey, $result)) {
              $this->icache[$ckey] = $result;
         }
@@ -861,6 +872,40 @@ class LDAP extends Net_LDAP3 {
         return $this->_list($base_dn, $filter, 'sub', $attributes, $search, $params);
     }
 
+    public function list_admins($attributes = array(), $domain = null)
+    {
+        if (empty($domain)) {
+            $domain = $this->domain;
+        }
+
+        $this->_log(LOG_DEBUG, "Auth::LDAP::list_admins(". var_export($attributes, true) . ", ".$domain.")");
+        $domaininfo = $this->domain_info($domain);
+        $result = array ('count' => 0, 'list' => array());
+        list($firstKey) = array_keys($domaininfo);
+        foreach ($domaininfo[$firstKey]['domainadmin'] as $domainadmin_dn) {
+             if ($domainadmin_dn == "cn=Directory Manager") {
+                 // don't show the Directory Manager in the list
+                 //$result['count']++;
+                 //$result['list'][$domainadmin_dn] = array('uid' => 'cn=Directory Manager', 'cn' => 'Directory Manager');
+             } else {
+                 $result['count']++;
+                 $userinfo = $this->user_info($domainadmin_dn, $attributes);
+                 foreach($userinfo as $domainadmin => $value) {
+                     $result['list'][$domainadmin] = $value;
+                     $domain = substr($domainadmin_dn, strpos($domainadmin_dn, ',ou=People,dc=') + strlen(',ou=People,dc='));
+                     $domain = str_replace(',dc=', '.', $domain);
+                     $result['list'][$domainadmin]['domain'] = $domain;
+                     $result['list'][$domainadmin]['tbitskolaboverallquota'] = $this->domainadmin_get_user_quota($domainadmin,'').
+                         ";".$result['list'][$domainadmin]['tbitskolaboverallquota'];
+                     $result['list'][$domainadmin]['tbitskolabmaxaccounts'] = $this->domainadmin_get_number_of_accounts($domainadmin).
+                         " / ".$result['list'][$domainadmin]['tbitskolabmaxaccounts'];
+                 }
+             }
+        }
+
+        return $result;
+    }
+
     public function organizationalunit_add($attrs, $typeid = null)
     {
         $base_dn = $this->entry_base_dn('ou', $typeid, $attrs);
@@ -1215,6 +1260,89 @@ class LDAP extends Net_LDAP3 {
         return $quota;
     }
 
+    private function collect_sieve_rules($result) {
+        $mail = $result['mail'];
+        if (is_array($mail)) {
+            $mail = $mail[0];
+        }
+
+        $ckey = "sieve_".$mail;
+
+        // use memcache
+        if ($result = $this->get_cache_data($ckey)) {
+            return $result;
+        }
+        // use internal cache
+        else if (isset($this->icache[$ckey])) {
+            return $this->icache[$ckey];
+        }
+
+        $sieve = new Net_Sieve();
+        $sieve_port = 4190;
+        $options = array('ssl' => array('verify_peer_name' => '', 'verify_peer' => '', 'allow_self_signed' => 1));
+     
+        $res = $sieve->connect('localhost', $sieve_port, $options, true);
+        $active_sieve_rules = '';
+        if (is_a($res, 'PEAR_Error')) {
+            $this->_log(LOG_ERROR, "cannot connect to sieve");
+            return $result;
+        }
+
+        $res = $sieve->login($this->conf->get('cyrus-imap', 'admin_login'), $this->conf->get('cyrus-imap', 'admin_password'), 'PLAIN', $mail);
+        if (is_a($res, 'PEAR_Error')) {
+            $this->_log(LOG_ERROR, "cannot login to sieve");
+            return $result;
+        }
+
+        $exts = $sieve->getExtensions();
+        $active = true;
+        $scripts = $sieve->listScripts($active);
+        foreach ($scripts as $scriptname) {
+            $script = $sieve->getScript($scriptname);
+            $script = new rcube_sieve_script($script, $exts);
+            foreach ($script->content as $content) {
+                if ($content['disabled'] == 1) {
+                    continue;
+                }
+//$this->_log(LOG_ERROR, print_r($content,true));
+
+                if (!array_key_exists('name', $content)) {
+                    continue;
+                }
+
+                $active_sieve_rules .= $content['name'].": ";
+                foreach ($content['actions'] as $action) {
+                    if ($action['type'] == 'include') {
+                        continue;
+                    }
+
+                    $active_sieve_rules .= $action['type'];
+
+                    if ($action['type'] == 'vacation') {
+                        $active_sieve_rules .= " '". $action['subject']. "'";
+                    } else if ($action['type'] == 'redirect') {
+                        $active_sieve_rules .= " '". $action['target']. "'";
+                    } else if ($action['type'] == 'notify') {
+                       $active_sieve_rules .= " '". $action['method']. "'";
+                    }
+
+                    $active_sieve_rules .= "; ";
+                }
+            }
+
+            $active_sieve_rules .= "<br/>";
+        }
+
+        $result['sieve'] = $active_sieve_rules;
+
+        // cache for one hour
+        if (!$this->set_cache_data($ckey, $quota, 1*60*60)) {
+             $this->icache[$ckey] = $active_sieve_rules;
+        }
+
+        return $result;
+    }
+
     public function user_info($user, $attributes = array('*'))
     {
         $this->_log(LOG_DEBUG, "Auth::LDAP::user_info() for user " . var_export($user, true));
@@ -1226,6 +1337,12 @@ class LDAP extends Net_LDAP3 {
             return false;
         }
 
+        if (in_array('sieve', $attributes)) {
+            // we need to exclude mail forwarding and domain admins, they don't have a sieve filter
+            $attributes[]='objectclass';
+            $attributes[]='mailforwardingaddress';
+        }
+
         $this->read_prepare($attributes);
 
         $result = $this->_read($user_dn, $attributes);
@@ -1234,6 +1351,11 @@ class LDAP extends Net_LDAP3 {
             $result[$user_dn]['tbitskolabquotaused'] = $this->collect_quota($result[$user_dn]);
         }
 
+        // check for sieve rules that are active: vacation and forwarding
+        if (in_array('sieve', $attributes) && in_array('mailrecipient', $result[$user_dn]['objectclass']) && empty($result[$user_dn]['mailforwardingaddress'])) {
+             $result[$user_dn] = $this->collect_sieve_rules($result[$user_dn]);
+        }
+
         return $result;
 
     }
diff --git a/lib/api/kolab_api_service_user.php b/lib/api/kolab_api_service_user.php
index 7b230d8..4c3394c 100644
--- a/lib/api/kolab_api_service_user.php
+++ b/lib/api/kolab_api_service_user.php
@@ -64,6 +64,11 @@ class kolab_api_service_user extends kolab_api_service
             $rights['find'] = "r";
         }
 
+        if (in_array('report', $effective_rights['entryLevelRights'])) {
+            $rights['userreport'] = "r";
+            $rights['adminreport'] = "r";
+        }
+
         $rights['effective_rights'] = "r";
 
         return $rights;
diff --git a/lib/api/kolab_api_service_users.php b/lib/api/kolab_api_service_users.php
index 73f3890..1a59a10 100644
--- a/lib/api/kolab_api_service_users.php
+++ b/lib/api/kolab_api_service_users.php
@@ -35,12 +35,20 @@ class kolab_api_service_users extends kolab_api_service
             'sn',
             'givenname',
             'mail',
+            'alias',
             'objectclass',
             'uidnumber',
             'gidnumber',
             'mailhost',
             'mailquota',
             'entrydn',
+            'tbitskolablastlogin',
+            'tbitskolabquotaused',
+            'tbitskolaboverallquota',
+            'tbitskolabdefaultquota',
+            'tbitskolabmaxaccounts',
+            'mailForwardingAddress',
+            'sieve',
         );
 
     /**
@@ -73,11 +81,26 @@ class kolab_api_service_users extends kolab_api_service
         $params = $this->parse_list_params($post);
         $search = $this->parse_list_search($post);
 
-        $users = $auth->list_users(null, $attributes, $search, $params);
-        $users = $this->parse_list_result($users);
+        if (in_array('tbitskolaboverallquota', $attributes)) {
+            // only get the domain admins
+            $users = $auth->list_admins(null, $attributes);
+            $users = $this->parse_list_result($users);
+        } else {
+            $users = $auth->list_users(null, $attributes, $search, $params);
+            $users = $this->parse_list_result($users);
+        }
 
         Log::trace("users.list result: " . var_export($users, TRUE));
 
+        if (in_array('tbitskolabquotaused', $attributes) || in_array('sieve', $attributes)) {
+            // need to query user_info for each user
+            foreach ($users['list'] as $uid_dn => $data) {
+                $detaildata = $auth->user_info($uid_dn, array('uid', 'tbitskolabquotaused', 'mailhost', 'mail', 'sieve'));
+                $users['list'][$uid_dn]['tbitskolabquotaused'] = $detaildata[$uid_dn]['tbitskolabquotaused'];
+                $users['list'][$uid_dn]['sieve'] = $detaildata[$uid_dn]['sieve'];
+            }
+        }
+
         return $users;
     }
 
diff --git a/lib/client/kolab_client_task_user.php b/lib/client/kolab_client_task_user.php
index 0faf286..02a713f 100644
--- a/lib/client/kolab_client_task_user.php
+++ b/lib/client/kolab_client_task_user.php
@@ -28,6 +28,8 @@ class kolab_client_task_user extends kolab_client_task
 
     protected $menu = array(
         'add'  => 'user.add',
+        'userreport' => 'user.userreport',
+        'adminreport' => 'user.adminreport',
     );
 
     protected $list_attribs = array('displayname', 'cn');
@@ -76,6 +78,134 @@ class kolab_client_task_user extends kolab_client_task
         $this->output->set_object('taskcontent', $output);
     }
 
+    public function action_userreport()
+    {
+        // only domain admins and users with role kolab-admin and the Directory Manager have access to this report
+        $output = "<table class='list'><tr><th>#_#report.uid</th><th>#_#report.name</th><th>#_#report.email</th><th>#_#report.forward</th><th>#_#report.quotausage</th><th>#_#report.lastlogin</th></tr>\n";
+
+        // Get all users within the current domain, and their quota usage and last login data and some more information
+        $result = $this->api_post('users.list', null, 
+                    array('attributes' => array('uid', 'mail', 'alias', 'sieve', 'cn', 'mailForwardingAddress', 'tbitskolablastlogin', 'tbitskolabquotaused')));
+        $userdata = $result->get();
+
+        foreach ($userdata['list'] as $longuid => $user) {
+
+            $name = $user['cn'];
+            $uid = $user['uid'];
+
+            if (is_array($user['mail'])) {
+                $email = "";
+                foreach ($user['mail'] as $mail) {
+                    if (!empty($email)) {
+                        $email .= ",<br/>";
+                    }
+                    $email .= $mail;
+                }
+            } else {
+                $email = $user['mail'];
+            }
+
+            if (is_array($user['alias'])) {
+                foreach ($user['alias'] as $alias) {
+                    if (!empty($email)) {
+                        $email .= ",<br/>";
+                    }
+                    $email .= $alias;
+                }
+            } else if (!empty($user['alias'])) {
+                if (!empty($email)) {
+                    $email .= ",<br/>";
+                }
+                $email .= $user['alias'];
+            }
+
+            $sieve = $user['sieve'];
+
+            if (is_array($user['mailforwardingaddress'])) {
+                $forward = "";
+                foreach ($user['mailforwardingaddress'] as $fwd) {
+                    if (!empty($forward)) {
+                        $forward .= ",<br/>";
+                    }
+                    $forward .= $fwd;
+                }
+            } else {
+                $forward = $user['mailforwardingaddress'];
+            }
+
+            if (!empty($user['tbitskolablastlogin'])) {
+                $lastlogin = date('d.m.Y H:i:s', $user['tbitskolablastlogin']);
+            } else {
+                $lastlogin = "";
+            }
+
+            // see https://github.com/TBits/KolabScripts/blob/master/kolab/patches/quotaused_wap.patch
+            if (!empty($user['mailforwardingaddress'])) {
+                $output .= "<tr class='selectable'><td>$uid</td><td>$name</td><td>$email</td><td>Mail forwarding to:</td><td colspan='2'>$forward</td></tr>\n";
+            } else if (empty($user['mail'])) {
+                // do not show domain admins in this list
+            } else {
+                $quota = kolab_html::inputquotaused(array('value' => $user['tbitskolabquotaused']));
+                $output .= "<tr class='selectable'><td>$uid</td><td>$name</td><td>$email</td><td>$sieve</td>".
+                    "<td style='text-align:center'>$quota</td>".
+                    "<td>$lastlogin</td></tr>\n";
+            }
+        }
+
+        $output .= "</table>\n";
+        $this->output->set_object('taskcontent', $output);
+        $this->output->add_translation('quota.unlimited');
+        $this->output->add_translation('quota.nomailbox');
+        $this->output->add_translation('report.uid');
+        $this->output->add_translation('report.name');
+        $this->output->add_translation('report.email');
+        $this->output->add_translation('report.forward');
+        $this->output->add_translation('report.quotausage');
+        $this->output->add_translation('report.lastlogin');
+    }
+
+    public function action_adminreport() {
+        // only domain admins and users with role kolab-admin and the Directory Manager have access to this report
+        $output = "<table class='list'><tr><th>#_#report.uid</th><th>#_#report.name</th><th>#_#report.accounts</th><th>#_#report.quotaassigned</th><th>#_#report.defaultquota</th><th>#_#report.lastlogin</th></tr>\n";
+
+        // Get all domain admins linked to the current domain, and their quota settings and last login data and some more information
+        $result = $this->api_post('users.list', null,
+                    array('attributes' => array('uid', 'cn', 'tbitskolaboverallquota', 'tbitskolabdefaultquota', 'tbitskolabmaxaccounts', 'tbitskolablastlogin', 'tbitskolabquotaused')));
+        $userdata = $result->get();
+
+        foreach ($userdata['list'] as $longuid => $user) {
+            $name = $user['cn'];
+            $uid = $user['uid'];
+            $domain = $user['domain'];
+
+            if (!empty($user['tbitskolablastlogin'])) {
+                $lastlogin = date('d.m.Y H:i:s', $user['tbitskolablastlogin']);
+            } else {
+                $lastlogin = "";
+            }
+
+            $overallquota = kolab_html::inputquotaused(array('value' => $user['tbitskolaboverallquota']));
+            $defaultquota = kolab_html::inputquotaused(array('value' => $user['tbitskolabdefaultquota']));
+            $output .= "<tr class='selectable'><td>$uid @ $domain</td><td>$name</td>".
+                "<td style='text-align:center'>".$user['tbitskolabmaxaccounts']."</td>".
+                "<td style='text-align:center'>".$overallquota."</td>".
+                "<td style='text-align:center'>".$defaultquota."</td>".
+                "<td>$lastlogin</td></tr>\n";
+        }
+
+        $output .= "</table>\n";
+        $this->output->set_object('taskcontent', $output);
+
+        $this->output->add_translation('quota.unlimited');
+        $this->output->add_translation('quota.nomailbox');
+        $this->output->add_translation('report.uid');
+        $this->output->add_translation('report.name');
+        $this->output->add_translation('report.accounts');
+        $this->output->add_translation('report.defaultquota');
+        $this->output->add_translation('report.quotaassigned');
+        $this->output->add_translation('report.lastlogin');
+    }
+
     private function user_form($attribs, $data = array())
     {
         if (empty($attribs['id'])) {
diff --git a/lib/locale/de_DE.php b/lib/locale/de_DE.php
index 5458494..7a8b33c 100644
--- a/lib/locale/de_DE.php
+++ b/lib/locale/de_DE.php
@@ -434,6 +434,17 @@ $LANG['user.postalcode'] = 'Postleitzahl';
 $LANG['user.postbox'] = 'Postfach';
 $LANG['user.postcode'] = 'Postleitzahl';
 $LANG['user.preferredlanguage'] = 'Muttersprache';
+$LANG['user.userreport'] = 'Übersicht über Benutzer';
+$LANG['user.adminreport'] = 'Administratoren';
+$LANG['report.uid'] = "Benutzername";
+$LANG['report.name'] = "Name";
+$LANG['report.email'] = "E-Mail";
+$LANG['report.forward'] = "Weiterleitungen/Urlaub";
+$LANG['report.defaultquota'] = "Quota Voreinstellung";
+$LANG['report.accounts'] = "Zugewiesene Benutzerkonten";
+$LANG['report.quotaassigned'] = "Zugewiesener Speicherplatz";
+$LANG['report.quotausage'] = "Speicherverbrauch";
+$LANG['report.lastlogin'] = "Letzter Login";
 $LANG['user.room'] = 'Raumnummer';
 $LANG['user.sn'] = 'Nachname';
 $LANG['user.street'] = 'Straße';
diff --git a/lib/locale/en_US.php b/lib/locale/en_US.php
index 8a459f3..4cf87df 100644
--- a/lib/locale/en_US.php
+++ b/lib/locale/en_US.php
@@ -435,6 +435,17 @@ $LANG['user.postalcode'] = 'Postal Code';
 $LANG['user.postbox'] = 'Postal box';
 $LANG['user.postcode'] = 'Postal code';
 $LANG['user.preferredlanguage'] = 'Native tongue';
+$LANG['user.userreport'] = 'Detailed Users List';
+$LANG['user.adminreport'] = 'Admins List';
+$LANG['report.uid'] = "UID";
+$LANG['report.name'] = "Name";
+$LANG['report.email'] = "E-Mail";
+$LANG['report.forward'] = "Forwards/Vacation";
+$LANG['report.defaultquota'] = "Default Quota";
+$LANG['report.accounts'] = "Assigned User accounts";
+$LANG['report.quotaassigned'] = "Quota Assigned";
+$LANG['report.quotausage'] = "Quota Usage";
+$LANG['report.lastlogin'] = "Last Login";
 $LANG['user.room'] = 'Room number';
 $LANG['user.sn'] = 'Surname';
 $LANG['user.statistics'] = 'Info';
