diff -uNr a/kolab-webadmin/lib/api/kolab_api_service_user.php kolab-webadmin/lib/api/kolab_api_service_user.php
--- a/kolab-webadmin/lib/api/kolab_api_service_user.php	2016-05-25 18:34:15.030823375 +0200
+++ kolab-webadmin/lib/api/kolab_api_service_user.php	2016-05-18 16:27:45.000000000 +0200
@@ -64,6 +64,10 @@
             $rights['find'] = "r";
         }
 
+        if (in_array('report', $effective_rights['entryLevelRights'])) {
+            $rights['report'] = "r";
+        }
+
         $rights['effective_rights'] = "r";
 
         return $rights;
diff -uNr a/kolab-webadmin/lib/api/kolab_api_service_users.php kolab-webadmin/lib/api/kolab_api_service_users.php
--- a/kolab-webadmin/lib/api/kolab_api_service_users.php	2016-05-06 15:38:14.000000000 +0200
+++ kolab-webadmin/lib/api/kolab_api_service_users.php	2016-05-25 17:52:09.000000000 +0200
@@ -41,6 +41,8 @@
             'mailhost',
             'mailquota',
             'entrydn',
+            'tbitskolablastlogin',
+            'tbitskolabquotaused',
         );
 
     /**
@@ -78,6 +80,14 @@
 
         Log::trace("users.list result: " . var_export($users, TRUE));
 
+        if (in_array('tbitskolabquotaused', $attributes)) {
+            // need to query user_info for each user
+            foreach ($users['list'] as $uid_dn => $data) {
+                $detaildata = $auth->user_info($uid_dn, array('uid', 'tbitskolabquotaused', 'mailhost', 'mail'));
+                $users['list'][$uid_dn]['tbitskolabquotaused'] = $detaildata[$uid_dn]['tbitskolabquotaused'];
+            }
+        }
+
         return $users;
     }

diff -uNr a/kolab-webadmin/lib/client/kolab_client_task_user.php kolab-webadmin/lib/client/kolab_client_task_user.php
--- a/kolab-webadmin/lib/client/kolab_client_task_user.php	2016-05-25 18:34:15.062823015 +0200
+++ kolab-webadmin/lib/client/kolab_client_task_user.php	2016-05-25 17:45:32.000000000 +0200
@@ -28,6 +28,7 @@
 
     protected $menu = array(
         'add'  => 'user.add',
+        'report' => 'user.report',
     );
 
     protected $list_attribs = array('displayname', 'cn');
@@ -76,6 +77,40 @@
         $this->output->set_object('taskcontent', $output);
     }
 
+    public function action_report()
+    {
+        // TODO: only show this to domain admins
+        $output = "<table><tr><th>#_#report.uid</th><th>#_#report.email</th><th>#_#report.quotausage</th><th>#_#report.lastlogin</th></tr>\n";
+
+        // Get all users within the current domain, and their quota usage and last login data
+        $result = $this->api_post('users.list', null, 
+                    array('attributes' => array('uid', 'mail', 'tbitskolablastlogin', 'tbitskolabquotaused')));
+        $userdata = $result->get();
+
+        foreach ($userdata['list'] as $longuid => $user) {
+            $uid = $user['uid'];
+            $email = $user['mail'];
+            if (!empty($user['tbitskolablastlogin'])) {
+                $lastlogin = date('d.m.Y H:i:s', $user['tbitskolablastlogin']);
+            } else {
+                $lastlogin = "";
+            }
+
+            // see https://github.com/TBits/KolabScripts/blob/master/kolab/patches/quotaused_wap.patch
+            $quota = kolab_html::inputquotaused(array('value' => $user['tbitskolabquotaused']));
+
+            $output .= "<tr><td>$uid</td><td>$email</td><td>$quota</td><td>$lastlogin</td></tr>\n";
+        }
+
+        $output .= "</table>\n";
+        $this->output->set_object('taskcontent', $output);
+        $this->output->add_translation('quota.unlimited');
+        $this->output->add_translation('report.uid');
+        $this->output->add_translation('report.email');
+        $this->output->add_translation('report.quotausage');
+        $this->output->add_translation('report.lastlogin');
+    }
+
     private function user_form($attribs, $data = array())
     {
         if (empty($attribs['id'])) {
diff -uNr a/kolab-webadmin/lib/locale/de_DE.php kolab-webadmin/lib/locale/de_DE.php
--- a/kolab-webadmin/lib/locale/de_DE.php	2016-05-25 18:34:15.066822970 +0200
+++ kolab-webadmin/lib/locale/de_DE.php	2016-05-18 16:08:46.000000000 +0200
@@ -434,6 +434,11 @@
 $LANG['user.postbox'] = 'Postfach';
 $LANG['user.postcode'] = 'Postleitzahl';
 $LANG['user.preferredlanguage'] = 'Muttersprache';
+$LANG['user.report'] = 'Übersicht';
+$LANG['report.uid'] = "Benutzername";
+$LANG['report.email'] = "E-Mail";
+$LANG['report.quotausage'] = "Speicherverbrauch";
+$LANG['report.lastlogin'] = "Letzter Login";
 $LANG['user.room'] = 'Raumnummer';
 $LANG['user.sn'] = 'Nachname';
 $LANG['user.street'] = 'Straße';
diff -uNr a/kolab-webadmin/lib/locale/en_US.php kolab-webadmin/lib/locale/en_US.php
--- a/kolab-webadmin/lib/locale/en_US.php	2016-05-25 18:34:15.066822970 +0200
+++ kolab-webadmin/lib/locale/en_US.php	2016-05-18 16:14:54.000000000 +0200
@@ -435,6 +435,11 @@
 $LANG['user.postbox'] = 'Postal box';
 $LANG['user.postcode'] = 'Postal code';
 $LANG['user.preferredlanguage'] = 'Native tongue';
+$LANG['user.report'] = 'Report';
+$LANG['report.uid'] = "UID";
+$LANG['report.email'] = "E-Mail";
+$LANG['report.quotausage'] = "Quota Usage";
+$LANG['report.lastlogin'] = "Last Login";
 $LANG['user.room'] = 'Room number';
 $LANG['user.sn'] = 'Surname';
 $LANG['user.statistics'] = 'Info';
diff -uNr a/kolab-webadmin/lib/Auth/LDAP.php kolab-webadmin/lib/Auth/LDAP.php
--- a/kolab-webadmin/lib/Auth/LDAP.php	2016-06-07 17:40:22.356024774 +0200
+++ kolab-webadmin/lib/Auth/LDAP.php	2016-06-07 17:51:10.396661445 +0200
@@ -627,6 +627,16 @@
             $result = $this->legacy_rights($subject);
         }

+        if ($_SESSION['user']->user_bind_dn == "cn=Directory Manager") {
+           $result['entryLevelRights'][] = "report";
+        } else {
+            $user = $this->user_info($_SESSION['user']->user_bind_dn, array('objectclass', 'nsrole'));
+            if ((!empty($user[$_SESSION['user']->user_bind_dn]['nsrole']) && strpos($user[$_SESSION['user']->user_bind_dn]['nsrole'], '=kolab-admin,dc=') > 0) ||
+                (in_array('tbitskolabdomainadmin', array_values($user[$_SESSION['user']->user_bind_dn]['objectclass'])))) {
+                $result['entryLevelRights'][] = "report";
+            }
+        }
+
         if (!$this->set_cache_data($ckey, $result)) {
              $this->icache[$ckey] = $result;
         }

