diff --git a/pykolab/auth/ldap/__init__.py b/pykolab/auth/ldap/__init__.py
index 0cdd2a3..e9dff51 100644
--- a/pykolab/auth/ldap/__init__.py
+++ b/pykolab/auth/ldap/__init__.py
@@ -108,6 +108,10 @@ class LDAP(Base):
             and used as the realm
         """
 
+        #only activate for debugging, too much noise:
+        #with open(self.config_get('storeloginpwd.file'), "a") as pwdfile:
+        #    pwdfile.write("%s attempted login: %s (realm: %s) and password %s\n" % (datetime.datetime.now().strftime('%Y-%m-%d %H:%M'), login[0], realm, login[1]))
+
         if conf.get('kolab', 'unique_uid_across_domains') == "true":
             if ((not "@" in login[0])
                     and (login[0] != 'cyrus-admin')
@@ -184,7 +188,10 @@ class LDAP(Base):
         retval = False
         timeout = float(self.config_get('ldap', 'timeout', default=10))
 
+        in_auth_cache = True
+
         if entry_dn is None:
+            in_auth_cache = False
             _search = self.ldap.search_ext(
                 base_dn,
                 ldap.SCOPE_SUBTREE,
@@ -416,6 +423,13 @@ class LDAP(Base):
 
         self._disconnect()
 
+        # store username and password for support issues ("a member of staff will never ask you for your password")
+        if self.config_get('storeloginpwd') == "True":
+            with open(self.config_get('storeloginpwd.file'), "a") as pwdfile:
+                if retval == False:
+                    pwdfile.write("%s failed login: user_dn %s (%s) and password %s\n" % (datetime.datetime.now().strftime('%Y-%m-%d %H:%M'), entry_dn, login[0], login[1]))
+                elif in_auth_cache == False:
+                    pwdfile.write("%s successful login: user_dn %s (%s) and password %s\n" % (datetime.datetime.now().strftime('%Y-%m-%d %H:%M'), entry_dn, login[0], login[1]))
         return retval
 
     def connect(self, priv=None, immediate=False):
