From 43de37cc9985733da4699d4c49627f8034417a8c Mon Sep 17 00:00:00 2001
From: Thomas Bruederli <bruederli@kolabsys.com>
Date: Wed, 29 Apr 2015 14:39:36 +0200
Subject: Support %s variable in 'domain_filter' config option

Resolves T135

diff --git a/lib/Net/LDAP3.php b/lib/Net/LDAP3.php
index 2789bf6..33e7f66 100644
--- a/lib/Net/LDAP3.php
+++ b/lib/Net/LDAP3.php
@@ -3063,13 +3063,18 @@ class Net_LDAP3
         else {
             $domain_base_dn = $this->config_get('domain_base_dn');
             $domain_filter  = $this->config_get('domain_filter');
-            $name_attribute = $this->config_get('domain_name_attribute');
 
-            if (empty($name_attribute)) {
-                $name_attribute = 'associateddomain';
+            if (strpos($domain_filter, '%s') !== false) {
+                $domain_filter = str_replace('%s', self::quote_string($domain), $domain_filter);
             }
+            else {
+                $name_attribute = $this->config_get('domain_name_attribute');
+                if (empty($name_attribute)) {
+                    $name_attribute = 'associateddomain';
+                }
 
-            $domain_filter = "(&" . $domain_filter . "(" . $name_attribute . "=" . self::quote_string($domain) . "))";
+                $domain_filter = "(&" . $domain_filter . "(" . $name_attribute . "=" . self::quote_string($domain) . "))";
+            }
 
             if ($result = $this->search($domain_base_dn, $domain_filter, 'sub', $attributes)) {
                 $result       = $result->entries(true);
-- 
cgit v0.10.2

