--- a/pykolab/auth/ldap/cache.py	2014-07-15 16:31:04.639728896 +0200
+++ b/pykolab/auth/ldap/cache.py	2014-07-15 16:35:48.239728529 +0200
@@ -52,7 +52,7 @@
 
 metadata = MetaData()
 
-db = None
+db = {}
 
 ##
 ## Classes
@@ -164,8 +164,8 @@
     """
     global db
 
-    if not db == None and not reinit:
-        return db
+    if domain in db and not reinit:
+        return db[domain]
 
     if reinit:
         import os
@@ -179,9 +179,9 @@
     metadata.create_all(engine)
 
     Session = sessionmaker(bind=engine)
-    db = Session()
+    db[domain] = Session()
 
-    return db
+    return db[domain]
 
 def last_modify_timestamp(domain):
     db = init_db(domain)
