--- a/pykolab/auth/ldap/__init__.py	2020-04-02 11:09:34.084689702 +0200
+++ b/pykolab/auth/ldap/__init__.py	2020-04-02 11:12:39.121105726 +0200
@@ -1931,7 +1931,8 @@
                     self.imap.user_mailbox_create(entry_changes[result_attribute])
 
                 elif not entry_changes[result_attribute] == old_canon_attr:
-                    self.imap.user_mailbox_rename(old_canon_attr, entry_changes[result_attribute])
+                    # do not rename an existing mailbox
+                    entry_changes[result_attribute] = old_canon_attr
 
         cache.get_entry(self.domain, entry)
 
@@ -2106,10 +2107,8 @@
                     )
 
                 else:
-                    self.imap.user_mailbox_rename(
-                        old_canon_attr,
-                        entry_changes[result_attribute]
-                    )
+                    # do not rename an existing mailbox
+                    entry_changes[result_attribute] = old_canon_attr
 
                 entry[result_attribute] = entry_changes[result_attribute]
                 cache.get_entry(self.domain, entry)
@@ -2121,10 +2120,8 @@
                     )
 
                 else:
-                    self.imap.user_mailbox_rename(
-                        old_canon_attr,
-                        entry[result_attribute]
-                    )
+                    # do not rename an existing mailbox
+                    entry[result_attribute] = old_canon_attr
 
                 cache.get_entry(self.domain, entry)
             else:
@@ -2290,10 +2287,8 @@
 
         if result_attribute in entry_changes and old_canon_attr is not None:
             if not entry_changes[result_attribute] == old_canon_attr:
-                self.imap.user_mailbox_rename(
-                    old_canon_attr,
-                    entry_changes[result_attribute]
-                )
+                # do not rename an existing mailbox
+                entry_changes[result_attribute] = old_canon_attr
 
         for key in entry_changes.keys():
             entry[key] = entry_changes[key]
