diff --git a/lib/Auth/LDAP.php b/lib/Auth/LDAP.php
index 4cda7cc..4173acb 100644
--- a/lib/Auth/LDAP.php
+++ b/lib/Auth/LDAP.php
@@ -271,7 +271,7 @@
         $admin_readonly_attrs = array("tbitsKolabMaxAccounts", "tbitsKolabOverallQuota");
 
         if (in_array('tbitsKolabUser', $this->classes_allowed())) {
-            $self_attrs = array_merge($self_attrs, array('tbitsKolabLastLogin'));
+            $self_attrs = array_merge($self_attrs, array('tbitsKolabLastLogin', 'tbitsKolabQuotaUsed'));
         }
 
         $_domain = str_replace('.', '_', $domain);
@@ -1102,6 +1102,8 @@ class LDAP extends Net_LDAP3 {
 
         $user_dn = key($user);
 
+        unset($attributes['tbitskolabquotaused']);
+
 #        if (empty($attributes['tbitskolabisdomainadmin'])) {
 #            $attributes['tbitskolabisdomainadmin'] = $user[$user_dn]['tbitskolabisdomainadmin'];
 #        }
@@ -1155,7 +1157,46 @@ class LDAP extends Net_LDAP3 {
 
         $this->read_prepare($attributes);
 
-        return $this->_read($user_dn, $attributes);
+        $result = $this->_read($user_dn, $attributes);
+
+        if (in_array('tbitskolabquotaused', $attributes) && in_array('mailhost', $attributes) && in_array('mail', $attributes)) {
+            $kolabhomeserver = $result[$user_dn]['mailhost'];
+            if (empty($kolabhomeserver)) {
+                $kolabhomeserver = "localhost";
+            }
+            // uri = imaps://localhost:993
+            $imapuri = $this->conf->get('cyrus-imap', 'uri');
+            $imap_port = "993";
+            $imap_manager = $this->conf->get('cyrus-imap', 'admin_login');
+            $imap_password = $this->conf->get('cyrus-imap', 'admin_password');
+
+            $imap_srv = "{" . $kolabhomeserver . ":" . $imap_port ."/imap/ssl/novalidate-cert}";
+            $mbox = imap_open($imap_srv, $imap_manager, $imap_password, OP_HALFOPEN);
+            if($mbox) {
+                $folder = "user/" . $result[$user_dn]['mail'];
+                $quota = @imap_get_quota($mbox, $folder);
+                if ($quota === false) {
+                    $boxes = imap_list($mbox, $imap_srv, '*');
+                    if( array_search($imap_srv.$folder, $boxes) === false ) {
+                        $quota = 'nomailbox';
+                    } else {
+                        $quota = 'unlimited';
+                    }
+                } else {
+                    $quota = $quota['usage'].";".$quota['limit'];
+                }
+                imap_close($mbox);
+            }
+            else {
+                // use 'unknown' value
+                $quota = 'unknown';
+            }
+
+            $result[$user_dn]['tbitskolabquotaused'] = $quota;
+        }
+
+        return $result;
+
     }
 
     public function user_find_by_attribute($attribute)
diff --git a/lib/client/kolab_client_task_settings.php b/lib/client/kolab_client_task_settings.php
index 18c8f96..e0e5af7 100644
--- a/lib/client/kolab_client_task_settings.php
+++ b/lib/client/kolab_client_task_settings.php
@@ -34,6 +34,7 @@ class kolab_client_task_settings extends kolab_client_task
     protected $form_element_types = array(
         'text', 'text-separated', 'text-quota', 'text-autocomplete',
         'text-unixtimestamp',
+        'text-quotaused',
         'select', 'multiselect',
         'list', 'list-autocomplete', 'checkbox', 'password', 'ldap_url',
         'aci', 'imap_acl',
diff --git a/lib/client/kolab_client_task_user.php b/lib/client/kolab_client_task_user.php
index df5aebe..0faf286 100644
--- a/lib/client/kolab_client_task_user.php
+++ b/lib/client/kolab_client_task_user.php
@@ -130,6 +130,7 @@ class kolab_client_task_user extends kolab_client_task
             'mailalternateaddress'      => 'contact_info',
 
             'tbitskolablastlogin'       => 'statistics',
+            'tbitskolabquotaused'       => 'statistics',
 
             /* POSIX Attributes first */
             'uid'                       => 'system',
diff --git a/lib/kolab_client_output.php b/lib/kolab_client_output.php
index f7bb471..820776d 100644
--- a/lib/kolab_client_output.php
+++ b/lib/kolab_client_output.php
@@ -127,7 +127,20 @@ class kolab_client_output
             $response['labels'][$label] = kolab_client_task::translate($label);
         }
 
-        return json_encode($response);
+        $response = json_encode($response);
+
+        $this->labels = array_unique($this->labels);
+        if (!empty($this->labels)) {
+            $l = array();
+            $t = array();
+            foreach ($this->labels as $label) {
+                $l[] = "#_#$label";
+                $t[] = kolab_client_task::translate($label);
+            }
+            $response = str_replace($l, $t, $response);
+        }
+
+        return $response;
     }
 
     /**
diff --git a/lib/kolab_client_task.php b/lib/kolab_client_task.php
index 137b8c0..d7c4dd7 100644
--- a/lib/kolab_client_task.php
+++ b/lib/kolab_client_task.php
@@ -914,6 +914,14 @@ class kolab_client_task
             $result['default'] = $field['default'];
             break;
 
+        case 'text-quotaused':
+            $result['type']    = kolab_form::INPUT_TEXTQUOTAUSED;
+            $result['default'] = $field['default'];
+
+            $this->output->add_translation('quota.nomailbox');
+            $this->output->add_translation('quota.unlimited');
+            break;
+
         case 'aci':
             $result['type']      = kolab_form::INPUT_TEXTAREA;
             $result['data-type'] = 'aci';
diff --git a/lib/kolab_form.php b/lib/kolab_form.php
index 8ebcda0..51a2ebf 100644
--- a/lib/kolab_form.php
+++ b/lib/kolab_form.php
@@ -39,6 +39,7 @@ class kolab_form
     const INPUT_CUSTOM   = 10;
     const INPUT_CONTENT  = 20;
     const INPUT_TEXTQUOTA = 30;
+    const INPUT_TEXTQUOTAUSED = 31;
     const INPUT_TEXTUNIXTIMESTAMP = 40;
 
     private $attribs  = array();
@@ -295,6 +296,11 @@ class kolab_form
             $content = kolab_html::inputquota($attribs);
             break;
 
+        case self::INPUT_TEXTQUOTAUSED:
+            $attribs['type'] = 'text';
+            $content = kolab_html::inputquotaused($attribs);
+            break;
+
         case self::INPUT_TEXTUNIXTIMESTAMP:
             $attribs['type'] = 'text';
             $content = kolab_html::inputunixtimestamp($attribs);
diff --git a/lib/kolab_html.php b/lib/kolab_html.php
index 911a239..e8596ad 100644
--- a/lib/kolab_html.php
+++ b/lib/kolab_html.php
@@ -209,6 +209,64 @@ class kolab_html
         return self::input($attribs) . self::select($select);
     }
 
+    public static function quota_to_read($valueInKB, $precise=false)
+    {
+        $result = array ('value' => $valueInKB, 'unit' => 'kb');
+        if ((!$precise || ($result['value'] % 1024 == 0)) && $result['value'] >= 1024) {
+            $result['value'] /= 1024;
+            $result['value'] = round($result['value']);
+            $result['unit'] = 'mb';
+        }
+        if ((!$precise || ($result['value'] % 1024 == 0)) && $result['value'] >= 1024) {
+            $result['value'] /= 1024;
+            $result['value'] = round($result['value'], 1);
+            $result['unit'] = 'gb';
+        }
+        return $result['value']." ".strtoupper($result['unit']);
+    }
+
+    /**
+     * Readonly control that will display the diskspace used compared to the booked quota
+     *
+     * @param array  $attribs  Element attributes
+     *
+     * @return string HTML output of the used quota
+     */
+    public static function inputquotaused($attribs = array())
+    {
+        $attribs['type'] = 'hidden';
+        $hidden_input = self::input($attribs);
+        unset($attribs['type']);
+        $attribs['name'] .= "_display";
+        $attribs['readonly'] = true;
+        $attribs['disabled'] = true;
+        if (!empty($attribs['value'])) {
+            if ($attribs['value'] == 'unknown') { 
+               $attribs['value'] = '';
+            } else if ($attribs['value'] == 'nomailbox') {
+               $attribs['value'] = '#_#quota.nomailbox';
+            } else if ($attribs['value'] == 'unlimited') {
+               $attribs['value'] = '#_#quota.unlimited';
+            } else if (strpos($attribs['value'], ';') !== false) {
+               $values = explode(';', $attribs['value']);
+               $quota = array();
+               $quota['usage'] = $values[0];
+               $quota['limit'] = $values[1];
+               $quota['percentage'] = round($quota['usage'] / $quota['limit'] * 100);
+               $quota['usage'] = self::quota_to_read($quota['usage']);
+               $quota['limit'] = self::quota_to_read($quota['limit'], true);
+               $attribs['value'] = $quota['usage']. " / ".$quota['limit']." (".$quota['percentage']."%)";
+               if ($quota['percentage'] >= 90) {
+                   $attribs['style'] = 'background-color:red';
+               }
+            } else {
+               $attribs['value'] = self::quota_to_read($attribs['value']);
+            }
+        }
+        $readonly_input = self::input($attribs);
+        return $hidden_input.$readonly_input;
+    }
+
     /**
      * Readonly control that will display the time encoded as a unix timestamp. Used for displaying tbitsKolabLastLogin
      *
diff --git a/lib/locale/de_DE.php b/lib/locale/de_DE.php
index e42ec2c..785208e 100644
--- a/lib/locale/de_DE.php
+++ b/lib/locale/de_DE.php
@@ -459,4 +459,7 @@ $LANG['user.tbitskolaboverallquota'] = 'Gesamtquota verfügbar';
 $LANG['user.tbitskolabdefaultquota'] = 'Voreinstellung Quota für Benutzerkonten';
 $LANG['user.statistics'] = 'Info';
 $LANG['user.tbitskolablastlogin'] = 'Letzte erfolgreiche Anmeldung';
+$LANG['user.tbitskolabquotaused'] = 'Aktueller Speicherplatzverbrauch';
+$LANG['quota.unlimited'] = "Unbegrenzt";
+$LANG['quota.nomailbox'] = "Noch kein Postfach vorhanden";
 
diff --git a/lib/locale/en_US.php b/lib/locale/en_US.php
index b8a9502..f02bdf8 100644
--- a/lib/locale/en_US.php
+++ b/lib/locale/en_US.php
@@ -446,6 +446,9 @@ $LANG['user.tbitskolabmaxaccounts'] = 'Maximum number of accounts';
 $LANG['user.tbitskolaboverallquota'] = 'Overall Quota assigned';
 $LANG['user.tbitskolabdefaultquota'] = 'Default Quota for user accounts';
 $LANG['user.tbitskolablastlogin'] = 'Latest successful login';
+$LANG['user.tbitskolabquotaused'] = 'Current quota usage';
+$LANG['quota.unlimited'] = "Unlimited";
+$LANG['quota.nomailbox'] = "Mailbox does not exist yet";
 $LANG['user.title'] = 'Job Title';
 $LANG['user.type_id'] = 'Account type';
 $LANG['user.uid'] = 'Unique identity (UID)';
