diff -uNr kolab-webadmin.orig/lib/Auth/LDAP.php kolab-webadmin/lib/Auth/LDAP.php
--- kolab-webadmin.orig/lib/Auth/LDAP.php	2016-04-14 09:39:03.000000000 +0200
+++ kolab-webadmin/lib/Auth/LDAP.php	2016-04-14 12:32:15.168909912 +0200
@@ -253,7 +253,7 @@
         $admin_readonly_attrs = array("tbitsKolabMaxAccounts", "tbitsKolabOverallQuota");
 
         if (in_array('tbitsKolabUser', $this->classes_allowed())) {
-            $self_attrs = array_merge($self_attrs, array('tbitsKolabLastLogin'));
+            $self_attrs = array_merge($self_attrs, array('tbitsKolabLastLogin', 'tbitsKolabQuotaUsed'));
         }
 
         $_domain = str_replace('.', '_', $domain);
@@ -1078,6 +1078,8 @@
 
         $user_dn = key($user);
 
+        unset($attributes['tbitskolabquotaused']);
+
 #      if (empty($attributes['tbitskolabisdomainadmin'])) {
 #            $attributes['tbitskolabisdomainadmin'] = $user[$user_dn]['tbitskolabisdomainadmin'];
 #        }
@@ -1131,7 +1133,37 @@
 
         $this->read_prepare($attributes);
 
-        return $this->_read($user_dn, $attributes);
+        $result = $this->_read($user_dn, $attributes);
+
+        if (in_array('tbitskolabquotaused', $attributes) && in_array('mailhost', $attributes) && in_array('mail', $attributes)) {
+            $kolabhomeserver = $result[$user_dn]['mailhost'];
+            // uri = imaps://localhost:993
+            $imapuri = $this->conf->get('cyrus-imap', 'uri');
+            $imap_port = "993";
+            $imap_manager = $this->conf->get('cyrus-imap', 'admin_login');
+            $imap_password = $this->conf->get('cyrus-imap', 'admin_password');
+
+            $mbox = imap_open("{" . $kolabhomeserver . ":" . $imap_port ."/imap/ssl/novalidate-cert}",
+                    $imap_manager, $imap_password, OP_HALFOPEN);
+            if($mbox) {
+                $quota = @imap_get_quota($mbox, "user/" . $result[$user_dn]['mail']);
+                imap_close($mbox);
+                if (empty($quota['limit'])) {
+                    $quota = 'unlimited';
+                } else {
+                    $quota = $quota['usage'].";".$quota['limit'];
+                }
+            }
+            else {
+                // use 'unknown' value
+                $quota = 'unknown';
+            }
+
+            $result[$user_dn]['tbitskolabquotaused'] = $quota;
+        }
+
+        return $result;
+
     }
 
     public function user_find_by_attribute($attribute)
diff -uNr kolab-webadmin.orig/lib/client/kolab_client_task_settings.php kolab-webadmin/lib/client/kolab_client_task_settings.php
--- kolab-webadmin.orig/lib/client/kolab_client_task_settings.php	2016-04-14 09:39:03.000000000 +0200
+++ kolab-webadmin/lib/client/kolab_client_task_settings.php	2016-04-14 09:48:28.351378654 +0200
@@ -34,6 +34,7 @@
     protected $form_element_types = array(
         'text', 'text-separated', 'text-quota', 'text-autocomplete',
         'text-unixtimestamp',
+        'text-quotaused',
         'select', 'multiselect',
         'list', 'list-autocomplete', 'checkbox', 'password', 'ldap_url',
         'aci', 'imap_acl',
diff -uNr kolab-webadmin.orig/lib/client/kolab_client_task_user.php kolab-webadmin/lib/client/kolab_client_task_user.php
--- kolab-webadmin.orig/lib/client/kolab_client_task_user.php	2016-04-14 09:39:03.000000000 +0200
+++ kolab-webadmin/lib/client/kolab_client_task_user.php	2016-04-14 09:59:11.820086212 +0200
@@ -130,6 +130,7 @@
             'mailalternateaddress'      => 'contact_info',
 
             'tbitskolablastlogin'       => 'statistics',
+            'tbitskolabquotaused'       => 'statistics',
 
             /* POSIX Attributes first */
             'uid'                       => 'system',
diff -uNr kolab-webadmin.orig/lib/kolab_client_task.php kolab-webadmin/lib/kolab_client_task.php
--- kolab-webadmin.orig/lib/kolab_client_task.php	2016-04-14 09:39:03.000000000 +0200
+++ kolab-webadmin/lib/kolab_client_task.php	2016-04-14 09:49:54.662400122 +0200
@@ -907,6 +907,13 @@
             $result['default'] = $field['default'];
             break;
 
+        case 'text-quotaused':
+            $result['type']    = kolab_form::INPUT_TEXTQUOTAUSED;
+            $result['default'] = $field['default'];
+
+            $this->output->add_translation('quota.unlimited');
+            break;
+
         case 'aci':
             $result['type']      = kolab_form::INPUT_TEXTAREA;
             $result['data-type'] = 'aci';
diff -uNr kolab-webadmin.orig/lib/kolab_form.php kolab-webadmin/lib/kolab_form.php
--- kolab-webadmin.orig/lib/kolab_form.php	2016-04-14 09:39:03.000000000 +0200
+++ kolab-webadmin/lib/kolab_form.php	2016-04-14 09:50:42.293860162 +0200
@@ -39,6 +39,7 @@
     const INPUT_CUSTOM   = 10;
     const INPUT_CONTENT  = 20;
     const INPUT_TEXTQUOTA = 30;
+    const INPUT_TEXTQUOTAUSED = 31;
     const INPUT_TEXTUNIXTIMESTAMP = 40;
 
     private $attribs  = array();
@@ -295,6 +296,11 @@
             $content = kolab_html::inputquota($attribs);
             break;
 
+        case self::INPUT_TEXTQUOTAUSED:
+            $attribs['type'] = 'text';
+            $content = kolab_html::inputquotaused($attribs);
+            break;
+
         case self::INPUT_TEXTUNIXTIMESTAMP:
             $attribs['type'] = 'text';
             $content = kolab_html::inputunixtimestamp($attribs);
diff -uNr kolab-webadmin.orig/lib/kolab_html.php kolab-webadmin/lib/kolab_html.php
--- kolab-webadmin.orig/lib/kolab_html.php	2016-04-14 09:39:03.000000000 +0200
+++ kolab-webadmin/lib/kolab_html.php	2016-04-14 12:30:04.566385693 +0200
@@ -209,6 +209,61 @@
         return self::input($attribs) . self::select($select);
     }
 
+    public static function quota_to_read($valueInKB, $precise=false)
+    {
+        $result = array ('value' => $valueInKB, 'unit' => 'kb');
+        if ((!$precise || ($result['value'] % 1024 == 0)) && $result['value'] >= 1024) {
+            $result['value'] /= 1024;
+            $result['value'] = round($result['value']);
+            $result['unit'] = 'mb';
+        }
+        if ((!$precise || ($result['value'] % 1024 == 0)) && $result['value'] >= 1024) {
+            $result['value'] /= 1024;
+            $result['value'] = round($result['value'], 1);
+            $result['unit'] = 'gb';
+        }
+        return $result['value']." ".strtoupper($result['unit']);
+    }
+
+    /**
+     * Readonly control that will display the diskspace used compared to the booked quota
+     *
+     * @param array  $attribs  Element attributes
+     *
+     * @return string HTML output of the used quota
+     */
+    public static function inputquotaused($attribs = array())
+    {
+        $attribs['type'] = 'hidden';
+        $hidden_input = self::input($attribs);
+        unset($attribs['type']);
+        $attribs['name'] .= "_display";
+        $attribs['readonly'] = true;
+        $attribs['disabled'] = true;
+        if (!empty($attribs['value'])) {
+            if ($attribs['value'] == 'unknown') { 
+               $attribs['value'] = '';
+            }
+            else if ($attribs['value'] == 'unlimited') {
+               $attribs['value'] = '_quota.unlimited';
+            } else {
+               $values = explode(';', $attribs['value']);
+               $quota = array();
+               $quota['usage'] = $values[0];
+               $quota['limit'] = $values[1];
+               $quota['percentage'] = round($quota['usage'] / $quota['limit'] * 100);
+               $quota['usage'] = self::quota_to_read($quota['usage']);
+               $quota['limit'] = self::quota_to_read($quota['limit'], true);
+               $attribs['value'] = $quota['usage']. " / ".$quota['limit']." (".$quota['percentage']."%)";
+               if ($quota['percentage'] >= 90) {
+                   $attribs['style'] = 'background-color:red';
+               }
+            }
+        }
+        $readonly_input = self::input($attribs);
+        return $hidden_input.$readonly_input;
+    }
+
     /**
      * Readonly control that will display the time encoded as a unix timestamp. Used for displaying tbitsKolabLastLogin
      *
diff -uNr kolab-webadmin.orig/lib/kolab_client_output.php kolab-webadmin/lib/kolab_client_output.php
--- kolab-webadmin.orig/lib/kolab_client_output.php	2015-01-21 02:01:36.000000000 +0100
+++ kolab-webadmin/lib/kolab_client_output.php	2016-05-27 17:37:35.279674625 +0200
@@ -127,7 +127,20 @@
             $response['labels'][$label] = kolab_client_task::translate($label);
         }
 
-        return json_encode($response);
+        $response = json_encode($response);
+
+        $this->labels = array_unique($this->labels);
+        if (!empty($this->labels)) {
+            $l = array();
+            $t = array();
+            foreach ($this->labels as $label) {
+                $l[] = "_$label";
+                $t[] = kolab_client_task::translate($label);
+            }
+            $response = str_replace($l, $t, $response);
+        }
+
+        return $response;
     }
 
     /**
diff -uNr kolab-webadmin.orig/lib/locale/de_DE.php kolab-webadmin/lib/locale/de_DE.php
--- kolab-webadmin.orig/lib/locale/de_DE.php	2016-04-14 09:39:03.000000000 +0200
+++ kolab-webadmin/lib/locale/de_DE.php	2016-04-14 14:13:57.791770256 +0200
@@ -459,4 +459,6 @@
 $LANG['user.tbitskolabdefaultquota'] = 'Voreinstellung Quota für Benutzerkonten';
 $LANG['user.statistics'] = 'Info';
 $LANG['user.tbitskolablastlogin'] = 'Letzte erfolgreiche Anmeldung';
+$LANG['user.tbitskolabquotaused'] = 'Aktueller Speicherplatzverbrauch';
+$LANG['quota.unlimited'] = "Unbegrenzt";
 
diff -uNr kolab-webadmin.orig/lib/locale/en_US.php kolab-webadmin/lib/locale/en_US.php
--- kolab-webadmin.orig/lib/locale/en_US.php	2016-04-14 09:39:03.000000000 +0200
+++ kolab-webadmin/lib/locale/en_US.php	2016-04-14 14:14:18.295537491 +0200
@@ -446,6 +446,8 @@
 $LANG['user.tbitskolaboverallquota'] = 'Overall Quota assigned';
 $LANG['user.tbitskolabdefaultquota'] = 'Default Quota for user accounts';
 $LANG['user.tbitskolablastlogin'] = 'Latest successful login';
+$LANG['user.tbitskolabquotaused'] = 'Current quota usage';
+$LANG['quota.unlimited'] = "Unlimited";
 $LANG['user.title'] = 'Job Title';
 $LANG['user.type_id'] = 'Account type';
 $LANG['user.uid'] = 'Unique identity (UID)';

