--- a/program/include/rcmail_output_html.php	2014-09-08 16:47:32.996201965 +0200
+++ b/program/include/rcmail_output_html.php	2014-09-08 16:48:01.008201926 +0200
@@ -184,7 +184,7 @@
             }
 
             $path = (rcube_utils::https_check() ? 'https' : 'http') . '://'
-                . $_SERVER['SERVER_NAME'] . rtrim($base, '/') . '/' . ltrim($path, '/');
+                . $_SERVER['HTTP_HOST'] . rtrim($base, '/') . '/' . ltrim($path, '/');
         }
 
         $this->assets_path = $path;
