--- a/plugins/libkolab/lib/kolab_storage_cache.php    2015-03-19 17:34:26.055006845 +0100
+++ b/plugins/libkolab/lib/kolab_storage_cache.php    2015-03-19 17:34:55.222006449 +0100
@@ -855,7 +855,7 @@
             $sql_data = $this->_serialize($object);

             // Skip multifolder insert for Oracle, we can't put long data inline
-            if ($this->db->db_provider == 'oracle') {
+            if (($this->db->db_provider == 'oracle') || ($this->db->db_provider == 'mysql')) {
                 $extra_cols = '';
                 if ($this->extra_cols) {
                     $extra_cols = array_map(function($n) { return "`{$n}`"; }, $this->extra_cols);
