--- a/public_html/assets/program/js/app.js	2016-11-29 15:39:49.589354082 +0100
+++ b/public_html/assets/program/js/app.js	2016-11-29 15:39:30.613543392 +0100
@@ -3912,6 +3912,8 @@
 
   this.open_compose_step = function(p)
   {
+    // this is helpful for my python tests, if the compose window takes longer to load
+    this.set_busy(true, 'loading');
     var url = this.url('mail/compose', p);
 
     // open new compose window

