diff -uNr a/pykolab/conf/__init__.py b/pykolab/conf/__init__.py
--- a/pykolab/conf/__init__.py	2014-06-26 10:27:16.443171000 +0200
+++ b/pykolab/conf/__init__.py	2014-06-26 10:27:56.139171001 +0200
@@ -278,6 +278,12 @@
                                     help    = _("Set the debugging " + \
                                         "verbosity. Maximum is 9, tracing " + \
                                         "protocols like LDAP, SQL and IMAP."))
+        
+        runtime_group.add_option(   "-e", "--default",
+                                    dest    = "answer_default",
+                                    action  = "store_true",
+                                    default = False,
+                                    help    = _("Use the default answer to all questions."))
 
         runtime_group.add_option(   "-l",
                                     dest    = "loglevel",
diff -uNr a/pykolab/utils.py b/pykolab/utils.py
--- a/pykolab/utils.py	2014-06-26 10:27:16.447171001 +0200
+++ b/pykolab/utils.py	2014-06-26 10:27:56.139171001 +0200
@@ -30,6 +30,7 @@
 from pykolab.translate import _
 
 log = pykolab.getLogger('pykolab.utils')
+conf = pykolab.getConf()
 
 def ask_question(question, default="", password=False, confirm=False):
     """
@@ -42,6 +43,12 @@
 
         Usage: pykolab.utils.ask_question("What is the server?", default="localhost")
     """
+    
+    if not default == "" and not default == None and conf.cli_keywords.answer_default:
+        if not conf.cli_keywords.quiet:
+            print ("%s [%s]: " % (question, default))
+        return default
+
     if password:
         if default == "" or default == None:
             answer = getpass.getpass("%s: " % (question))
@@ -108,6 +115,14 @@
         default_no = "'no'"
         default_yes = "Please type 'yes'"
 
+    if conf.cli_keywords.answer_yes or (conf.cli_keywords.answer_default and default_answer is not None):
+        if not conf.cli_keywords.quiet:
+            print ("%s [%s/%s]: " % (question,default_yes,default_no))
+        if conf.cli_keywords.answer_yes:
+            return True
+        if conf.cli_keywords.answer_default:
+            return default_answer
+
     answer = False
     while answer == False:
         answer = raw_input("%s [%s/%s]: " % (question,default_yes,default_no))
@@ -129,6 +144,11 @@
                 return True
 
 def ask_menu(question, options={}, default=''):
+    if not default == '' and conf.cli_keywords.answer_default:
+        if not conf.cli_keywords.quiet:
+            print question + " [" + default + "]:"
+        return default
+
     if not default == '':
         print question + " [" + default + "]:"
     else:
@@ -282,6 +302,9 @@
 def multiline_message(message):
     _msg = ""
 
+    if conf.cli_keywords.quiet:
+        return ""
+
     column_width = 80
 
     # First, replace all occurences of "\n"
