diff --git a/lib/api/kolab_api_service_form_value.php b/lib/api/kolab_api_service_form_value.php
--- a/lib/api/kolab_api_service_form_value.php
+++ b/lib/api/kolab_api_service_form_value.php
@@ -1136,6 +1136,9 @@
                 if (!$this->_validate_email_address_in_any_of_my_domains($mail_address)) {
                     throw new Exception("Email address '$mail_address' not in local domain", 693);
                 }
+                if (!$this->_validate_crossdomain_alias($mail_address, $postdata['mail'])) {
+                    throw new Exception("Alias '$mail_address' must be configured manually for '".$postdata['mail']."'");
+                }
             }
         }
 
@@ -1483,4 +1483,51 @@
         return $valid;
     }
 
+    private function _validate_crossdomain_alias($mail_address, $primary_address)
+    {
+        $at_index = strrpos($mail_address, "@");
+        if (is_bool($at_index) && !$at_index) {
+            throw new Exception("Invalid email address: No domain name space", 235);
+        } else {
+            $email_domain = substr($mail_address, $at_index+1);
+        }
+
+        $my_primary_domain = $_SESSION['user']->get_domain();
+
+        if ($email_domain == $my_primary_domain) {
+            return true;
+        }
+
+        $valid = true;
+
+        // if /etc/postfix/ldap/virtual_alias_maps.cf contains: search_base = dc=%2,dc=%1
+        // instead of the hard coded domain name then we need a row 
+        // in the manually edited virtual alias file that matches the two domains
+        $conf = Conf::get_instance();
+        $postfix_virtual_file = $conf->get("kolab", "postfix_virtual_file");
+        if ($postfix_virtual_file != null) {
+            $valid = false;
+            $localpart_alias=substr($mail_address, 0, $at_index);
+            $localpart_primary_address=substr($primary_address, 0, strrpos($primary_address, "@"));
+            $content = file($postfix_virtual_file);
+            foreach ($content as $line) {
+                if (trim($line) == $mail_address.' '.$primary_address) {
+                    // catchall, or specific address
+                    $valid = true;
+                } else if ($localpart_alias == $localpart_primary_address) {
+                    // only forward one address, to the same email address local part
+                    if (trim($line) == '@'.$email_domain.' @' .$my_primary_domain) {
+                        $valid = true;
+                    }
+                }
+            }
+        }
+
+        if (!$valid) {
+            Log::trace("Found email address to be in one of my domains but the link to ".$my_primary_domain." is missing in ".$postfix_virtual_file);
+        }
+
+        return $valid;
+    }
+
 }
