diff --git a/public_html/js/kolab_admin.js b/public_html/js/kolab_admin.js
index 5059ce5..db84744 100644
--- a/public_html/js/kolab_admin.js
+++ b/public_html/js/kolab_admin.js
@@ -1160,9 +1160,18 @@ function kolab_admin()
           else {
             var focused = selected[e.which == 40 ? 'next' : 'prev']();
 
-            while (focused.length && focused.is(':hidden'))
+            while (focused.length && focused.is(':hidden')) {
+              var prev = focused[0].firstChild.data;
               focused = selected[e.which == 40 ? 'next' : 'prev']();
 
+              // test if there is only one visible element
+              if (prev == focused[0].firstChild.data)
+              {
+                focused = options['first']();
+                break;
+              }
+            }
+
             if (!focused.length)
               focused = options[e.which == 40 ? 'first' : 'last']();
 
