--- /tmp/kolab-webadmin/public_html/js/kolab_admin.js	2013-04-16 15:50:26.898544067 +0200
+++ /usr/share/kolab-webadmin/public_html/js/kolab_admin.js	2013-07-02 09:49:36.906113977 +0200
@@ -1056,9 +1056,18 @@
           else {
             var focused = selected[e.which == 40 ? 'next' : 'prev']();
 
-            while (focused.length && focused.is(':hidden'))
+            while (focused.length && focused.is(':hidden')) {
+              var prev = focused[0].firstChild.data;
               focused = selected[e.which == 40 ? 'next' : 'prev']();
 
+              // test if there is only one visible element
+              if (prev == focused[0].firstChild.data)
+              {
+                focused = options['first']();
+                break;
+              }
+            }
+
             if (!focused.length)
               focused = options[e.which == 40 ? 'first' : 'last']();

