diff --git a/lib/api/kolab_api_service_form_value.php b/lib/api/kolab_api_service_form_value.php
index ddccbfe..02f362a 100644
--- a/lib/api/kolab_api_service_form_value.php
+++ b/lib/api/kolab_api_service_form_value.php
@@ -1152,10 +1152,22 @@ class kolab_api_service_form_value extends kolab_api_service
         }
     }
 
-    private function validate_mailquota($value)
+    private function validate_mailquota($value, $postdata)
     {
-        //return (int)($value);
-        return $value;
+        $conf = Conf::get_instance();
+        $quota_attribute = $conf->get('quota_attribute');
+
+        if ($postdata[$quota_attribute.'-unit'] == 'gb') {
+            $postdata[$quota_attribute] *= 1024*1024;
+            $postdata[$quota_attribute.'-unit'] = 'kb';
+        }
+
+        if ($postdata[$quota_attribute.'-unit'] == 'mb') {
+            $postdata[$quota_attribute] *= 1024;
+            $postdata[$quota_attribute.'-unit'] = 'kb';
+        }
+
+        return $postdata[$quota_attribute];
     }
 
     private function validate_mailalternateaddress($value)
diff --git a/lib/client/kolab_client_task_settings.php b/lib/client/kolab_client_task_settings.php
index 121b85e..ffe9e09 100644
--- a/lib/client/kolab_client_task_settings.php
+++ b/lib/client/kolab_client_task_settings.php
@@ -34,6 +34,7 @@ class kolab_client_task_settings extends kolab_client_task
     protected $form_element_types = array(
         'text', 'select', 'multiselect', 'list', 'list-autocomplete', 'checkbox', 'password', 'ldap_url',
         'list-domainadmins',
+        'text-quota',
     );
 
 
diff --git a/lib/kolab_client_task.php b/lib/kolab_client_task.php
index 2b10e42..6677564 100644
--- a/lib/kolab_client_task.php
+++ b/lib/kolab_client_task.php
@@ -854,6 +854,10 @@ class kolab_client_task
             $result['data-type'] = kolab_form::TYPE_LIST;
             break;
 
+        case 'text-quota':
+            $result['type'] = kolab_form::INPUT_TEXTQUOTA;
+            break;
+
         default:
             $result['type'] = kolab_form::INPUT_TEXT;
             if (isset($field['maxlength'])) {
diff --git a/lib/kolab_form.php b/lib/kolab_form.php
index d8ddfe6..125dab5 100644
--- a/lib/kolab_form.php
+++ b/lib/kolab_form.php
@@ -38,6 +38,7 @@ class kolab_form
     const INPUT_HIDDEN   = 9;
     const INPUT_CUSTOM   = 10;
     const INPUT_CONTENT  = 20;
+    const INPUT_TEXTQUOTA = 30;
     const INPUT_DOMAINADMIN = 40;
 
     const TYPE_LIST = 1;
@@ -278,6 +279,11 @@ class kolab_form
             $content = kolab_html::input($attribs);
             break;
 
+        case self::INPUT_TEXTQUOTA:
+            $attribs['type'] = 'text';
+            $content = kolab_html::inputquota($attribs);
+            break;
+
         case self::INPUT_CHECKBOX:
             $attribs['type'] = 'checkbox';
             $content = kolab_html::input($attribs);
diff --git a/lib/kolab_html.php b/lib/kolab_html.php
index 33bc705..c0ef529 100644
--- a/lib/kolab_html.php
+++ b/lib/kolab_html.php
@@ -154,6 +154,33 @@ class kolab_html
     }
 
     /**
+     * Input element for mail quota. user can select the unit (GB, MB)
+     *
+     * @param array  $attribs  Element attributes
+     *
+     * @return string HTML output of the input
+     */
+    public static function inputquota($attribs = array())
+    {
+        $elem_attribs = array_merge(self::$input_attribs, self::$input_event_attribs,
+            self::$common_attribs, self::$event_attribs);
+        $selected_option_mb = $selected_option_gb = "";
+        $selected_option_kb = "selected";
+        if ($attribs['value'] >= 1024 && $attribs['value'] % 1024 == 0) {
+            $attribs['value'] /= 1024;
+            $selected_option_kb = "";
+            $selected_option_mb = "selected";
+        }
+        if ($attribs['value'] >= 1024 && $attribs['value'] % 1024 == 0) {
+            $attribs['value'] /= 1024;
+            $selected_option_mb = "";
+            $selected_option_gb = "selected";
+        }
+
+        return sprintf('<input%s /><select name="%s-unit"><option value="kb" %s>KB</option><option value="mb" %s>MB</option><option value="gb" %s>GB</option></select>', self::attrib_string($attribs, $elem_attribs), $attribs['name'], $selected_option_kb, $selected_option_mb, $selected_option_gb);
+    }
+
+    /**
      * Textarea element.
      *
      * @param array $attribs  Element attributes
