diff --git a/lib/kolab_recipient_policy.php b/lib/kolab_recipient_policy.php
index 15d190a..a946d29 100644
--- a/lib/kolab_recipient_policy.php
+++ b/lib/kolab_recipient_policy.php
@@ -125,6 +125,10 @@ class kolab_recipient_policy {
             $userdata['domain'] = $_SESSION['user']->get_domain();
         }
 
+	if ($primary_mail == '') {
+		$primary_mail = $conf->get_raw('primary_mail');
+	}
+
         preg_match_all('/%\((\w+)\)s/', $primary_mail, $substrings);
 
         // Update userdata array
