--- kolab_auth.php.orig	2013-04-15 12:07:43.000000000 +0200
+++ kolab_auth.php	2013-04-15 11:10:27.605569812 +0200
@@ -32,6 +32,7 @@
 {
     private $ldap;
     private $data = array();
+    private $base_dn="";
 
     public function init()
     {
@@ -258,6 +259,10 @@
     {
         $this->load_config();
 
+        $user=$args['user'];
+        $domain=substr($user,strpos($user, '@') + 1);
+        $this->base_dn="dc=".implode(",dc=",explode(".", $domain));
+
         if (!$this->init_ldap()) {
             $args['abort'] = true;
             return $args;
@@ -437,6 +442,10 @@
      */
     private function init_ldap()
     {
+        if ($this->base_dn == "") {
+            return null;
+        }
+        
         if ($this->ldap) {
             return $this->ldap->ready;
         }
@@ -453,6 +462,10 @@
         if (empty($addressbook)) {
             return false;
         }
+        
+        // for supporting multiple domains
+        $addressbook['base_dn'] = "ou=People,".$this->base_dn;
+        $addressbook['groups']['base_dn'] = "ou=Groups,".$this->base_dn;
 
         $this->ldap = new kolab_auth_ldap_backend(
             $addressbook,
