diff --git a/lib/api/kolab_api_service_form_value.php b/lib/api/kolab_api_service_form_value.php
index ddccbfe..43a6453 100644
--- a/lib/api/kolab_api_service_form_value.php
+++ b/lib/api/kolab_api_service_form_value.php
@@ -225,12 +225,15 @@ class kolab_api_service_form_value extends kolab_api_service
                 $result[$attr_name] = $this->{$method_name}($attr_value, $postdata);
             } elseif (array_key_exists($attr_name, $attribs['form_fields']) && !empty($attribs['form_fields'][$attr_name]['validate']) && !$attribs['form_fields'][$attr_name]['validate']) {
                 $result[$attr_name] = $attr_value;
-            } else {
+            } else if (empty($attr_value)) {
                 try {
                     $result[$attr_name] = $this->{$method_name}($attr_value, $postdata);
                 } catch (Exception $e) {
-                    Log::debug("Attribute $attr_name did not validate, but it is not a required attribute. Not saving. (Error was: $e)");
+                    Log::debug("Attribute $attr_name did not validate, but it is not a required attribute. Saving empty value. (Error was: $e)");
+                    $result[$attr_value] = $attr_value;
                 }
+            } else {
+                $result[$attr_name] = $this->{$method_name}($attr_value, $postdata);
             }
 
         }
